---
title: "Import data"
author: "`r authors`"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
---

# Import data

This section shows how to import the _biom_ and its accompanying data
files to a _TreeSummarizedExperiment_ object. A TSE object is a standard data structure 
that is utilized in the miaverse.

We use example data from the following publication: Tengeler AC, Dam
SA, Wiesmann M, Naaijen J, van Bodegom M, Belzer C, Dederen PJ,
Verweij V, Franke B, Kozicz T, Vasquez AA & Kiliaan AJ (2020) [**Gut
microbiota from persons with attention-deficit/hyperactivity disorder
affects the brain in
mice**](https://doi.org/10.1186/s40168-020-00816-x). Microbiome 8:44.

In this study, mice are colonized with microbiota from participants
with ADHD (attention deficit hyperactivity disorder) and healthy
participants.  The aim of the study was to assess whether the mice
display ADHD behaviors after being inoculated with ADHD microbiota,
suggesting a role of the microbiome in ADHD pathology.

## Data

The Data is located in a "data" subfolder. It consists of 3 files:

-   biom file: contains an abundance table and taxonomy information
-   csv file: contains the sample metadata
-   tree file: contains a phylogenetic tree.

Define source file paths.

```{r}
biom_file_path <- "data/Aggregated_humanization2.biom"
sample_meta_file_path <- "data/Mapping_file_ADHD_aggregated.csv"
tree_file_path <- "data/Data_humanization_phylo_aggregation.tre"
```  

Load the (biom) data into a SummarizedExperiment (SE) object.

```{r}
se <- loadFromBiom(biom_file_path)
```  


## Investigate the R data object

We have now imported the data set in R. Let us investigate its contents.

```{r}
print(se)
```


The `assays` slot includes a list of abundance tables. The imported abundance table is named as "counts".
Let us inspect only the first cols and rows.

```{r}
assays(se)$counts[1:3, 1:3]
```


### rowData (taxonomic information)

The `rowdata` includes taxonomic information from the biom file. The `head()` command
shows just the beginning of the data table for an overview.

`knitr::kable()` is for printing the information more nicely.

```{r}
knitr::kable(head(rowData(se)))
```

These taxonomic rank names (column names) are not real rank names. Letâ€™s replace them with real rank names.

In addition to that, the taxa names include, e.g., '"k__' before the name, so let's
make them cleaner by removing them. 

```{r}
names(rowData(se)) <- c("Kingdom", "Phylum", "Class", "Order", 
                        "Family", "Genus")

# Goes through the whole DataFrame. Removes '.*[kpcofg]__' from strings, where [kpcofg] 
# is any character from listed ones, and .* any character.
rowdata_modified <- BiocParallel::bplapply(rowData(se), 
                                           FUN = stringr::str_remove, 
                                           pattern = '.*[kpcofg]__')

# Genus level has additional '\"', so let's delete that also
rowdata_modified <- BiocParallel::bplapply(rowdata_modified, 
                                           FUN = stringr::str_remove, 
                                           pattern = '\"')

# rowdata_modified is a list, so it is converted back to DataFrame format. 
rowdata_modified <- DataFrame(rowdata_modified)

# And then assigned back to the SE object
rowData(se) <- rowdata_modified

# Now we have a nicer table
knitr::kable(head(rowData(se)))
```

### colData (sample information)

We notice that the imported biom file did not contain the sample meta data
yet, so it includes an empty data frame.

```{r}
head(colData(se))
```


Let us add a sample meta data file.

```{r}
# We use this to check what type of data it is
# read.table(sample_meta_file_path)

# It seems like a comma separated file and it does not include headers
# Let us read it and then convert from data.frame to DataFrame
# (required for our purposes)
sample_meta <- DataFrame(read.table(sample_meta_file_path, sep = ",", header = FALSE))

# Add sample names to rownames
rownames(sample_meta) <- sample_meta[,1]

# Delete column that included sample names
sample_meta[,1] <- NULL

# We can add headers
colnames(sample_meta) <- c("patient_status", "cohort", "patient_status_vs_cohort", "sample_name")

# Then it can be added to colData
colData(se) <- sample_meta
```

Now `colData` includes the sample metadata. Use kable to print it more nicely.

```{r}
knitr::kable(head(colData(se)))
```


### Phylogenetic tree information

Now, let's add a phylogenetic tree.

The current data object, se, is a SummarizedExperiment object. This
does not include a slot for adding a phylogenetic tree. In order to do
this, we can convert the SE object to an extended TreeSummarizedExperiment
object which also includes a `rowTree` slot.


```{r}
tse <- as(se, "TreeSummarizedExperiment")

# tse includes same data as se
print(tse)
```

Next, let us read the tree data file and add it to the R data object (tse).


```{r}
tree <- ape::read.tree(tree_file_path)

# Add tree to rowTree
rowTree(tse) <- tree

# Check
tse
```

Now `rowTree` includes a phylogenetic tree:

```{r, eval=FALSE}
head(rowTree(tse))
```