---
title: "Differential abundance analysis"
output: html_notebook
---

## TODO
- Polish code
- Polish text

#################################################

In this notebook, we analyse abundances with two different methods: *-test, and DESeq2.
Both of these tests test, if there are statistical significant difference between 
groups.

### T-test
T-test is  parametric, so it requires normally distributed data. 
It is always wise to check if the data is normally distributed. If it is not then
the data should be normalized. If that is not possible then non-parametric equivalents 
(t-test = Mannâ€“Whitney U test should be used.

- Make CLR transformation for the data, then run t-test for every ASV (in a for loop for instance; see t.test() ) between two groups (pick any grouping variable in the metadata, just for demonstration; we can change that later easily if needed) and pick p values t.test(...)$p.value; then do multiple testing correction with p.adjust(), then plot histogram of p-values and adjusted p-values; you can check [this code](https://github.com/microbiome/microbiome/blob/master/inst/extdata/check_wilcoxon.R) for some ideas

Let's first collect the data that we are going to use in t-test.
```{r}
# Does clr transformation. Pseudocount is added, because data contains zeros, and
# clr transformation includes log transformation.
tse <- transformCounts(tse, method = "clr", pseudocount = 1)

# Does transpose, so samples are in rows, then creates a data frame.
abundance_analysis_data <- data.frame(t(assay(tse, "clr")))

# Then we need variable for grouping samples. "Diet" column includes diet information.
# There two groups "western" and "control". Let's include that to the data frame.
abundance_analysis_data <- cbind(abundance_analysis_data, Groups = colData(tse)$Diet)

# Subsets data if, more than 2 different groups
# abundance_analysis_data_sub <- abundance_analysis_data[abundance_analysis_data$Groups == "KO_t1_western" |
#                                                          abundance_analysis_data$Groups == "WT_t2_western",]

```

Next, let's check the normality of data. It can be done with Shapiro-Wilk test.
As we can see, data is not normally distributed.

I don't know if normality test is necessary.##################################

```{r}
# Does Shapiro-Wilk test. Does it only for columns that contain abundances, not for
# column that contain Groups.

normality_test_p <- c()

for (column in 
     abundance_analysis_data[, !names(abundance_analysis_data) %in% "Groups"]){
  # Does Shapiro-Wilk test
  result <- shapiro.test(column)
  
  # Stores p-value to vector
  normality_test_p <- c(normality_test_p, result$p.value)
}

print(paste0("P-values over 0.05: ", sum(normality_test_p>0.05), "/", sum(normality_test_p<=0.05)))
```

Now we can do the Student's t-test. We test all the taxa by looping through columns,
and store individual p-values to a vector. Then we create a data frame from collected
data.

```{r}
# Does t-test. Does it only for columns that contain abundances, not for
# column that contain Groups.

t_test_p <- c()

for (column in 
     names(abundance_analysis_data[, !names(abundance_analysis_data) %in% "Groups"])){
  #print(column)
  result <- t.test(abundance_analysis_data[,column] ~ Groups, data = abundance_analysis_data)
   
   # Stores p-value to a vector
   t_test_p  <- c(t_test_p , result$p.value)
}

t_test_p <- data.frame(taxa = 
                         names(abundance_analysis_data[, !names(abundance_analysis_data) 
                                                       %in% "Groups"]),
                       p_raw = t_test_p)

```

Because of multiple tests were made, we need to adjust p-values. That is because, 
these individual tests are not independent. Because of that, if enough tests are made
statistically we will get positive result, even though positive result is extremely 
rare, lets say, e.g., 1 %. 

To avoid false positive results, we must adjust p-values, when multiply tests are made
from same data. With adjustment, the significance levels of individual tests are stricter. 
Here we use Holm method, but there are several other methods also.

```{r}
# Does the adjustment
t_test_p$p_holm <- p.adjust(t_test_p$p_raw, method = "holm")
```

```{r}

df = data.frame(x = c(t_test_p$p_raw, t_test_p$p_holm), 
               type=rep(c("raw", "holm"), c(length(t_test_p$p_raw), length(t_test_p$p_holm))))

t_test_plot <- ggplot(d) + geom_histogram(aes(x=x, fill=type)) +
  xlab("p-value") + 
  ylab("Frequency")

t_test_plot
```

- Then add, for the same data, [DESeq2](https://github.com/microbiome/tutorials/blob/master/deseq2.Rmd) example with our present data (exclude ANOVA from that example code)

### DESeq2
Our second analysis method is DESeq2. It does automatically normalization, and the
analysis. It also takes care the p-value adjustment, so we don't have to worry 
about that.

DESeq2 is a package for gene expression analysis. It offers automatic analysis of variation.
It uses binomial distribution to detect expression differences between groups. 
Because it uses normalization, it takes care the difference between library sizes, 
and differences between library composition. In addition to gene analysis, 
DESeq2 can also be used to other purposes.

DESeq2 analysis includes multiple steps, but they are done automatically
1. Raw counts are normalized by log-based scaling.
2. Taxa-wise variance is estimated. Values tell how much each taxa varies between samples.
3. Curve is fitted over all those taxa-wise variance estimates that we got in the last step.
This model tells how big is the variance in specific abundance level. 
4. The model is used to shrink those individual variance estimates to avoid the effect of, 
e.g., small sample size and higher variance. This reduce the likelihood to get 
false positives.
5. Variance estimates are used to compare different groups, and we got the result
if variance is explained by groups.

```{r, echo=FALSE}
library(DESeq2)

# Running the DESeq2 analysis
# Creates DESeq2 object from the data. Uses "Diet" to create groups. 
ds2 <- DESeqDataSet(tse, ~Diet)
# Does the analysis
dds <- DESeq(ds2)
# Gets the results from the object
res <- results(dds)
# Creates a data frame from results
df <- as.data.frame(res)
# Adds taxon colmun that includes names of taxa
df$taxon <- rownames(df)
# Orders the rows of data frame in increasing order firstly based on column "log2FoldChange" 
# and secondly based on "padj" column
df <- df %>% dplyr::arrange(log2FoldChange, padj)

print(head(knitr::kable((df))))

```

### Comparison of t-test and DESeq2
Let's compare results that we got from the t-test and DESeq2.
As we can see from the scatter plot, most of the values are the same. 
- Add scatterplot that compares the p-values from both methods

```{r}
mf <- data.frame(df$padj, t_test_p$p_holm)
p <- ggplot(mf, aes(x = df$padj, y = t_test_p$p_holm)) +
       labs(x = "DESeq2 adjusted p-value", y = "t-test adjusted p-value") +
       geom_count()+
 scale_size_area(max_size = 10)

print(p)
```

