---
title: "Differential abundance analysis"
output: html_notebook
---

In this notebook, we analyse abundances with two different methods: **t-test**, and **DESeq2**.
Both of these tests test, if there are statistical significant difference between 
groups.

### T-test
T-test tests if there is difference between two groups. It is parametric, so it 
requires normally distributed data. It is always wise to check if the data is normally 
distributed. If it is not then the data should be normalized. If that is not possible 
then non-parametric equivalents (t-test = Mannâ€“Whitney U) test should be used.

Let's first collect the data that we are going to use in t-test.

```{r}
# Does clr transformation. Pseudocount is added, because data contains zeros, and
# clr transformation includes log transformation.
tse <- transformCounts(tse, method = "clr", pseudocount = 1)

# Does transpose, so samples are in rows, then creates a data frame.
abundance_analysis_data <- data.frame(t(assay(tse, "clr")))

# Then we need variable for grouping samples. "patient_status" column includes information
# about patients' status. There two groups "ADHD" and "control". Let's include that to the data frame.
abundance_analysis_data <- cbind(abundance_analysis_data, patient_status = colData(tse)$patient_status)

# Subsets data if, more than 2 different groups
# abundance_analysis_data_sub <- abundance_analysis_data[
#abundance_analysis_data$patient_status_vs_cohort == "ADHD_Cohort_1" |
#abundance_analysis_data$patient_status_vs_cohort == "Control_Cohort_3",]
```

Next, let's check the normality of data. It can be done with Shapiro-Wilk test.
As we can see, data is not normally distributed.

I don't know if normality test is necessary.###################################

```{r}
# Does Shapiro-Wilk test. Does it only for columns that contain abundances, not for
# column that contain Groups.

normality_test_p <- c()

for (column in 
     abundance_analysis_data[, !names(abundance_analysis_data) %in% "patient_status"]){
  # Does Shapiro-Wilk test
  result <- shapiro.test(column)
  
  # Stores p-value to vector
  normality_test_p <- c(normality_test_p, result$p.value)
}

print(paste0("P-values over 0.05: ", sum(normality_test_p>0.05), "/", length(normality_test_p)))
```

Now we can do the Student's t-test. We test all the taxa by looping through columns,
and store individual p-values to a vector. Then we create a data frame from collected
data.

```{r}
# Does t-test. Does it only for columns that contain abundances, not for
# column that contain patient status.

t_test_p <- c()

for (column in 
     names(abundance_analysis_data[, !names(abundance_analysis_data) %in% "patient_status"])){
  #print(column)
  result <- t.test(abundance_analysis_data[,column] ~ patient_status, data = abundance_analysis_data)
   
   # Stores p-value to a vector
   t_test_p  <- c(t_test_p , result$p.value)
}

t_test_p <- data.frame(taxa = 
                         names(abundance_analysis_data[, !names(abundance_analysis_data) 
                                                       %in% "patient_status"]),
                       p_raw = t_test_p)
```

As multiple tests were performed and these tests are not independent, we need to adjust p-values. Otherwise, we would increase the chance of a type I error drastically depending on our p-value threshold. By applying p-value adjustment, we can keep the false positive rate at a level that is acceptable. What is acceptable depends on our research goals. Here we use the Holm method, but there are several other methods as well.

```{r}
# Does the adjustment
t_test_p$p_holm <- p.adjust(t_test_p$p_raw, method = "holm")
```

```{r}

df = data.frame(x = c(t_test_p$p_raw, t_test_p$p_holm), 
               type=rep(c("raw", "holm"), c(length(t_test_p$p_raw), length(t_test_p$p_holm))))

t_test_plot <- ggplot(df) + geom_histogram(aes(x=x, fill=type)) +
  xlab("p-value") + 
  ylab("Frequency")

t_test_plot
```

### DESeq2
Our second analysis method is DESeq2. It performs the data normalization  automatically. It also takes care of the p-value adjustment, so we don't have to worry 
about that.

DESeq2 is a package for gene expression analysis. It offers automatic analysis of variation.
It uses the binomial distribution to detect differences in gene expression between groups. 
Because it uses normalization, it takes care the difference between library sizes, 
and differences between library composition. In addition to gene analysis, 
DESeq2 can also be used for other purposes. DESeq2 analysis includes multiple steps, but they are done automatically:  

1. Raw counts are normalized by log-based scaling.  
2. Taxa-wise variance is estimated. These values tell how much each taxa varies between samples.  
3. A curve is fitted over all those taxa-wise variance estimates that we got in the last step.  
This model tells how big the variance is in a specific abundance level. 
4. The model is used to shrink those individual variance estimates to avoid the effect of, 
e.g., small sample size and higher variance. This reduces the likelihood to get 
false positives.  
5. Variance estimates are used to compare different groups. We receive a result that shows whether the variance is explained by groups.  


More information can be found, e.g., from Harvard Chan Bioinformatic Core's tutorial
[Introduction to DGE - ARCHIVED](https://hbctraining.github.io/DGE_workshop/lessons/04_DGE_DESeq2_analysis.html)

```{r, results='hide'}
library(DESeq2)
library(dplyr)
```


```{r}
# Running the DESeq2 analysis

# Creates DESeq2 object from the data. Uses "patient_status" to create groups. 
ds2 <- DESeqDataSet(tse, ~patient_status)
# Does the analysis
dds <- DESeq(ds2)
# Gets the results from the object
res <- results(dds)
# Creates a data frame from results
df <- as.data.frame(res)
# Adds taxon column that includes names of taxa
df$taxon <- rownames(df)
# Orders the rows of data frame in increasing order firstly based on column "log2FoldChange" 
# and secondly based on "padj" column
df <- df %>% arrange(log2FoldChange, padj)

print(head(knitr::kable((df))))

```

### Comparison of t-test and DESeq2
Let's compare results that we got from the t-test and DESeq2.
As we can see from the scatter plot, DESeq2 gives lower p-values than t-test. 

```{r}
mf <- data.frame(df$padj, t_test_p$p_holm)
p <- ggplot(mf, aes(x = df$padj, y = t_test_p$p_holm)) +
       labs(x = "DESeq2 adjusted p-value", y = "t-test adjusted p-value") +
       geom_count()+
 scale_size_area(max_size = 10)

print(p)
```

```{r}
# Prints number of p-values under 0.05
print(paste0("DESeq2 p-values under 0.05: ", sum(df$padj<0.05, na.rm = TRUE), "/", length(df$padj)))
print(paste0("T-test p-values under 0.05: ", sum(t_test_p$p_holm<0.05, na.rm = TRUE), "/", length(t_test_p$p_holm)))
```

### Comparison of abundance

In previous steps, we got information which taxa vary between ADHD and control groups.
Let's plot those taxa in the boxplot, and compare visually if abundances of those taxa
differ in ADHD and control samples. For comparison, let's plot also taxa that do not
differ between ADHD and control groups. 

```{r}
# Sorts p-values in increasing order. Takes 3rd first ones. Takes those rows that match
# with p-values. Takes taxa. 
highest3 <- df[df$padj %in% sort(df$padj, decreasing = FALSE)[1:3], ]$taxon
# From clr transformed table, takes only those taxa that had highest p-values
highest3 <-assay(tse, "clr")[highest3, ]
# Transposes the table
highest3 <- t(highest3)
# Adds colData that includes patient status infomation
highest3 <- data.frame(highest3, as.data.frame(colData(tse)))

# Sorts p-values in decreasing order. Takes 3rd first ones. Takes those rows that match
# with p-values. Takes taxa. 
lowest3 <- df[df$padj %in% sort(df$padj, decreasing = TRUE)[1:3], ]$taxon
# From clr transformed table, takes only those taxa that had lowest p-values
lowest3 <-assay(tse, "clr")[lowest3, ]
# Transposes the table
lowest3 <- t(lowest3)
# Adds colData that includes patient status infomation
lowest3 <- data.frame(lowest3, as.data.frame(colData(tse)))
```

```{r}
# Puts plots in the same picture
gridExtra::grid.arrange(
  
  # Plot 1
  ggplot(highest3, aes(x = patient_status, y = highest3[,1])) + 
    geom_boxplot() + 
    ylab("CLR abundances") + # y axis title
    ggtitle(names(highest3)[1]) + # main title
    theme(title = element_text(size = 10),
          axis.text = element_text(size = 12),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title
  
  # Plot 2
  ggplot(highest3, aes(x = patient_status, y = highest3[,2])) + 
    geom_boxplot() + 
    ylab("CLR abundances") + # y axis title
    ggtitle(names(highest3)[2]) + # main title
    theme(title = element_text(size = 10),
          axis.text = element_text(size = 12),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title
  
  # Plot 3
  ggplot(highest3, aes(x = patient_status, y = highest3[,3])) + 
    geom_boxplot() + 
    ylab("CLR abundances") + # y axis title
    ggtitle(names(highest3)[3]) + # main title
    theme(title = element_text(size = 10),
          axis.text = element_text(size = 12),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title
  
  # Plot 4
  ggplot(lowest3, aes(x = patient_status, y = lowest3[,1])) + 
    geom_boxplot() + 
    ylab("CLR abundances") + # y axis title
    ggtitle(names(lowest3)[1]) + # main title
    theme(title = element_text(size = 10),
          axis.text = element_text(size = 12),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title
  
  # Plot 5
  ggplot(lowest3, aes(x = patient_status, y = lowest3[,2])) + 
    geom_boxplot() + 
    ylab("CLR abundances") + # y axis title
    ggtitle(names(lowest3)[2]) + # main title
    theme(title = element_text(size = 10),
          axis.text = element_text(size = 12),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title
  
  # Plot 6
  ggplot(lowest3, aes(x = patient_status, y = lowest3[,3])) + 
    geom_boxplot() + 
    ylab("CLR abundances") + # y axis title
    ggtitle(names(lowest3)[3]) + # main title
    theme(title = element_text(size = 10),
          axis.text = element_text(size = 12),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title
  
  # 3 columns and 2 rows
  ncol = 3, 
  nrow = 2
)
```
