---
title: "Import data"
output: html_notebook
---

This notebook imports _biom_ data to a _TreeSummarizedExperiment_ object. 

Data is located in a "data" subfolder. Data consists of 3 files: 

- biom file that contains abundance table and taxonomy information
- txt file that contains sample metadata
- nwk file that contains a phylogenetic tree.

## Initialization

[Install](install.R) the necessary R packages if you have not already
done it.

Then load the R packages.

```{r}
library("mia")
```

Define source file paths.

```{r}
biom_file_path <- "data/table-with-taxonomy.biom"
sample_meta_file_path <- "data/WesternDietSert_extendedTimeComparisonsv4.txt"
tree_file_path <- "data/tree.nwk"
```


Create SE object from biom file and investigate it.

```{r}
se <- loadFromBiom(biom_file_path)
```


## Investigate the R data object

We have now imported the data in R. Let us investigate its contents.

```{r}
print(se)
```


Counts include the abundance table from biom file. Let us just use first cols and rows.

```{r}
assays(se)$counts[1:3, 1:3]
```

Now rowdata includes taxonomical information from biom file. The head() command
shows just the beginning of the data table for an overview.

```{r}
knitr::kable(head(rowData(se)))
```

Taxonomic ranks are not real rank names. Let's replace those taxonomic classes 
with real rank names. 

In addition to that, taxa names include, e.g., "D_0__" before the name, so let's
make them cleaner by removing them. 

```{r}
names(rowData(se)) <- c("Confidence", "Kingdom", "Phylum", "Class", "Order", 
                        "Family", "Genus", "Species")

# Goes through whole DataFrame. Removes "D_[0-9]__" from strings, where [0-9]] 
# is any number between 0 and 9. 
rowdata_modified <- BiocParallel::bplapply(rowData(se), 
                                           FUN = stringr::str_remove, 
                                           pattern = "D_[0-9]__")

# rowdata_modified is list, so it is converted back to DataFrame. 
rowdata_modified <- DataFrame(rowdata_modified)

# And then assigned back to the SE object
rowData(se) <- rowdata_modified

knitr::kable(head(rowData(se)))

```

We notice that the imported biom file did not contain sample meta data
yet, so it includes empty data frame

```{r}
head(colData(se))
```

## Add side information

Let us add sample meta data file

```{r}
# We use this to check what type of data is it
# read.table(sample_meta_file_path)

# It seems like a comma separated file and includes headers
# Let us read it and then convert from data.frame to DataFrame
# (required for our purposes)
sample_meta <- 
sample_meta <- DataFrame(read.table(sample_meta_file_path, sep = ",", header = TRUE))

# Then it can be added to colData
colData(se) <- sample_meta
```

Now colData includes sample metadata. Use kable to print it more nicely.

```{r}
knitr::kable(head(colData(se)))
```

Now, let's add a phylogenetic tree.

```{r}
# Current object, se, is a SummarizedExperiment object, thus, it does not include
# slot for phylogenetic tree. However, we can convert SE object to 
# TreeSummarizedExperiment which includes rowTree slot.
tse <- as(se, "TreeSummarizedExperiment")

# tse includes same data as se
print(tse)
```

```{r}
# Reads the tree 
tree <- ape::read.tree(tree_file_path)

# Adds tree to rowTree
rowTree(tse) <- tree

# Check
head(tse)
```

Now rowTree includes phylogenetic tree:

```{r, eval=FALSE}
head(rowTree(tse))
```

### TODO

The rowData columns are not real taxonomic ranks (Phylum, Genus..);
replace the column names with the real rank names for D0, D1 etc. Do
we already have a function for this (D0, D1 etc are standard and
should have a direct conversion).

- Yes, it causes an error in next steps, if they are not real ranks. I try to find better way. 

(Link to data exploration / to next step?)

