<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Importing microbiome data | Questions for Lab 1, 2 and 3</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Importing microbiome data | Questions for Lab 1, 2 and 3" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/course_2021_radboud" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Importing microbiome data | Questions for Lab 1, 2 and 3" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="Leo lahti &amp; Rajesh Shigdel" />


<meta name="date" content="2022-01-10" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="reproducible-reporting-with-rmarkdown.html"/>
<link rel="next" href="microbiome-data-exploration.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to microbiome data science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#learning-goals"><i class="fa fa-check"></i><b>1.2</b> Learning goals</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="program.html"><a href="program.html"><i class="fa fa-check"></i><b>2</b> Program</a>
<ul>
<li class="chapter" data-level="2.1" data-path="program.html"><a href="program.html#day-1-from-raw-sequences-to-ecological-data-analysis"><i class="fa fa-check"></i><b>2.1</b> Day 1: from raw sequences to ecological data analysis</a></li>
<li class="chapter" data-level="2.2" data-path="program.html"><a href="program.html#day-2---alpha-diversity"><i class="fa fa-check"></i><b>2.2</b> Day 2 - Alpha diversity</a></li>
<li class="chapter" data-level="2.3" data-path="program.html"><a href="program.html#day-3---beta-diversity"><i class="fa fa-check"></i><b>2.3</b> Day 3 - Beta diversity</a></li>
<li class="chapter" data-level="2.4" data-path="program.html"><a href="program.html#day-4--differential-abundance"><i class="fa fa-check"></i><b>2.4</b> Day 4- Differential abundance</a></li>
<li class="chapter" data-level="2.5" data-path="program.html"><a href="program.html#day-5-presentations-closing"><i class="fa fa-check"></i><b>2.5</b> Day 5 : Presentations &amp; closing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#checklist-before-the-course"><i class="fa fa-check"></i><b>3.1</b> Checklist (before the course)</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#support-and-resources"><i class="fa fa-check"></i><b>3.2</b> Support and resources</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#installing-and-loading-the-required-r-packages"><i class="fa fa-check"></i><b>3.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="reproducible-reporting-with-rmarkdown.html"><a href="reproducible-reporting-with-rmarkdown.html"><i class="fa fa-check"></i><b>4</b> Reproducible reporting with Rmarkdown</a></li>
<li class="chapter" data-level="5" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html"><i class="fa fa-check"></i><b>5</b> Importing microbiome data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#data-access"><i class="fa fa-check"></i><b>5.1</b> Data access</a></li>
<li class="chapter" data-level="5.2" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#importing-microbiome-data-in-r"><i class="fa fa-check"></i><b>5.2</b> Importing microbiome data in R</a></li>
<li class="chapter" data-level="5.3" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#example-solutions"><i class="fa fa-check"></i><b>5.3</b> Example solutions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html"><i class="fa fa-check"></i><b>6</b> Microbiome data exploration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#data-structure"><i class="fa fa-check"></i><b>6.1</b> Data structure</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#transformations"><i class="fa fa-check"></i><b>6.1.1</b> Transformations</a></li>
<li class="chapter" data-level="6.1.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#aggregation"><i class="fa fa-check"></i><b>6.1.2</b> Aggregation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#visualization"><i class="fa fa-check"></i><b>6.2</b> Visualization</a></li>
<li class="chapter" data-level="6.3" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#exercises-optional"><i class="fa fa-check"></i><b>6.3</b> Exercises (optional)</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>7</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#visualization-1"><i class="fa fa-check"></i><b>7.1</b> Visualization</a></li>
<li class="chapter" data-level="7.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#statistical-testing-and-comparisons"><i class="fa fa-check"></i><b>7.2</b> Statistical testing and comparisons</a></li>
<li class="chapter" data-level="7.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#exercises"><i class="fa fa-check"></i><b>7.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#examples-of-pcoa-with-different-settings"><i class="fa fa-check"></i><b>8.1</b> Examples of PCoA with different settings</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-bray-curtis"><i class="fa fa-check"></i><b>8.1.1</b> PCoA for ASV-level data with Bray-Curtis</a></li>
<li class="chapter" data-level="8.1.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-aitchison-distance"><i class="fa fa-check"></i><b>8.1.2</b> PCoA for ASV-level data with Aitchison distance</a></li>
<li class="chapter" data-level="8.1.3" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-aggregated-to-phylum-level"><i class="fa fa-check"></i><b>8.1.3</b> PCoA aggregated to Phylum level</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#highlighting-external-variables"><i class="fa fa-check"></i><b>8.2</b> Highlighting external variables</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#discrete-grouping-variable-shown-with-colors"><i class="fa fa-check"></i><b>8.2.1</b> Discrete grouping variable shown with colors</a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-plot-with-continuous-variable"><i class="fa fa-check"></i><b>8.2.2</b> PCoA plot with continuous variable</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#estimating-associations-with-an-external-variable"><i class="fa fa-check"></i><b>8.3</b> Estimating associations with an external variable</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#community-typing"><i class="fa fa-check"></i><b>8.4</b> Community typing</a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#exercises-1"><i class="fa fa-check"></i><b>8.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html"><i class="fa fa-check"></i><b>9</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#wilcoxon-test"><i class="fa fa-check"></i><b>9.1</b> Wilcoxon test</a></li>
<li class="chapter" data-level="9.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#ancom-bc"><i class="fa fa-check"></i><b>9.2</b> ANCOM-BC</a></li>
<li class="chapter" data-level="9.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>9.3</b> Comparison of the methods</a></li>
<li class="chapter" data-level="9.4" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-abundance"><i class="fa fa-check"></i><b>9.4</b> Comparison of abundance</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="study-material.html"><a href="study-material.html"><i class="fa fa-check"></i><b>10</b> Study material</a>
<ul>
<li class="chapter" data-level="10.1" data-path="study-material.html"><a href="study-material.html#lecture-slides"><i class="fa fa-check"></i><b>10.1</b> Lecture slides</a></li>
<li class="chapter" data-level="10.2" data-path="study-material.html"><a href="study-material.html#r-programming-resources"><i class="fa fa-check"></i><b>10.2</b> R programming resources</a></li>
<li class="chapter" data-level="10.3" data-path="study-material.html"><a href="study-material.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>10.3</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="10.4" data-path="study-material.html"><a href="study-material.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>10.4</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="10.5" data-path="study-material.html"><a href="study-material.html#further-reading"><i class="fa fa-check"></i><b>10.5</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="additional-community-typing.html"><a href="additional-community-typing.html"><i class="fa fa-check"></i><b>11</b> Additional Community Typing</a>
<ul>
<li class="chapter" data-level="11.1" data-path="additional-community-typing.html"><a href="additional-community-typing.html#community-composition"><i class="fa fa-check"></i><b>11.1</b> Community composition</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="additional-community-typing.html"><a href="additional-community-typing.html#composition-barplot"><i class="fa fa-check"></i><b>11.1.1</b> Composition barplot</a></li>
<li class="chapter" data-level="11.1.2" data-path="additional-community-typing.html"><a href="additional-community-typing.html#composition-heatmap"><i class="fa fa-check"></i><b>11.1.2</b> Composition heatmap</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="additional-community-typing.html"><a href="additional-community-typing.html#cluster-into-csts"><i class="fa fa-check"></i><b>11.2</b> Cluster into CSTs</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="additional-community-typing.html"><a href="additional-community-typing.html#elbow-method"><i class="fa fa-check"></i><b>11.2.1</b> Elbow Method</a></li>
<li class="chapter" data-level="11.2.2" data-path="additional-community-typing.html"><a href="additional-community-typing.html#silhouette-method"><i class="fa fa-check"></i><b>11.2.2</b> Silhouette Method</a></li>
<li class="chapter" data-level="11.2.3" data-path="additional-community-typing.html"><a href="additional-community-typing.html#gap-statistic-method"><i class="fa fa-check"></i><b>11.2.3</b> Gap-Statistic Method</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="additional-community-typing.html"><a href="additional-community-typing.html#holmes-and-mcmurdie-workflow"><i class="fa fa-check"></i><b>11.3</b> Holmes and McMurdie Workflow</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="additional-community-typing.html"><a href="additional-community-typing.html#jensen-shannon-distance"><i class="fa fa-check"></i><b>11.3.1</b> Jensen-Shannon Distance</a></li>
<li class="chapter" data-level="11.3.2" data-path="additional-community-typing.html"><a href="additional-community-typing.html#nonmetric-multidimensional-scaling"><i class="fa fa-check"></i><b>11.3.2</b> NonMetric Multidimensional Scaling</a></li>
<li class="chapter" data-level="11.3.3" data-path="additional-community-typing.html"><a href="additional-community-typing.html#chi-squared-correspondence-results"><i class="fa fa-check"></i><b>11.3.3</b> Chi-Squared Correspondence Results</a></li>
<li class="chapter" data-level="11.3.4" data-path="additional-community-typing.html"><a href="additional-community-typing.html#bray-curtis-mds"><i class="fa fa-check"></i><b>11.3.4</b> Bray-Curtis MDS</a></li>
<li class="chapter" data-level="11.3.5" data-path="additional-community-typing.html"><a href="additional-community-typing.html#bray-curtis-nmds"><i class="fa fa-check"></i><b>11.3.5</b> Bray-Curtis NMDS</a></li>
<li class="chapter" data-level="11.3.6" data-path="additional-community-typing.html"><a href="additional-community-typing.html#other-distances-from-the-philentropy-package"><i class="fa fa-check"></i><b>11.3.6</b> Other distances from the Philentropy package</a></li>
<li class="chapter" data-level="11.3.7" data-path="additional-community-typing.html"><a href="additional-community-typing.html#add-a-clustering-variable"><i class="fa fa-check"></i><b>11.3.7</b> Add a Clustering Variable</a></li>
<li class="chapter" data-level="11.3.8" data-path="additional-community-typing.html"><a href="additional-community-typing.html#more-clustering"><i class="fa fa-check"></i><b>11.3.8</b> More Clustering</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="additional-community-typing.html"><a href="additional-community-typing.html#cst-analysis"><i class="fa fa-check"></i><b>11.4</b> CST Analysis</a></li>
<li class="chapter" data-level="11.5" data-path="additional-community-typing.html"><a href="additional-community-typing.html#session-info"><i class="fa fa-check"></i><b>11.5</b> Session Info</a></li>
<li class="chapter" data-level="11.6" data-path="additional-community-typing.html"><a href="additional-community-typing.html#bibliography"><i class="fa fa-check"></i><b>11.6</b> Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="exercise-solutions.html"><a href="exercise-solutions.html"><i class="fa fa-check"></i><b>12</b> Exercise Solutions</a>
<ul>
<li class="chapter" data-level="12.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#section-5"><i class="fa fa-check"></i><b>12.1</b> Section 5</a></li>
<li class="chapter" data-level="12.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#section-6"><i class="fa fa-check"></i><b>12.2</b> Section 6</a></li>
<li class="chapter" data-level="12.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#section-7"><i class="fa fa-check"></i><b>12.3</b> Section 7</a></li>
<li class="chapter" data-level="12.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#section-8"><i class="fa fa-check"></i><b>12.4</b> Section 8</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="tutorial-aims.html"><a href="tutorial-aims.html"><i class="fa fa-check"></i><b>13</b> Tutorial Aims</a></li>
<li class="chapter" data-level="14" data-path="lab-1.html"><a href="lab-1.html"><i class="fa fa-check"></i><b>14</b> Lab 1</a>
<ul>
<li class="chapter" data-level="14.1" data-path="lab-1.html"><a href="lab-1.html#unsupervised-learning-feature-selection-dimension-reduction"><i class="fa fa-check"></i><b>14.1</b> unsupervised learning: Feature selection &amp; dimension reduction</a></li>
<li class="chapter" data-level="14.2" data-path="lab-1.html"><a href="lab-1.html#load-the-data"><i class="fa fa-check"></i><b>14.2</b> Load the data</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="lab-2.html"><a href="lab-2.html"><i class="fa fa-check"></i><b>15</b> Lab 2</a>
<ul>
<li class="chapter" data-level="15.1" data-path="lab-2.html"><a href="lab-2.html#unsupervised-learning-clustering-visualization"><i class="fa fa-check"></i><b>15.1</b> unsupervised learning: clustering &amp; visualization</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="lab-3.html"><a href="lab-3.html"><i class="fa fa-check"></i><b>16</b> Lab 3</a>
<ul>
<li class="chapter" data-level="16.1" data-path="lab-3.html"><a href="lab-3.html#unsupervised-learninganalysis-and-interpretation"><i class="fa fa-check"></i><b>16.1</b> unsupervised learning:Analysis and interpretation</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Questions for Lab 1, 2 and 3</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importing-microbiome-data" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Importing microbiome data</h1>
<p>This section demonstrates how to import microbiome profiling data in R.</p>
<div id="data-access" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Data access</h2>
<p><strong>Option 1</strong></p>
<p><em>ADHD-associated changes in gut microbiota and brain in a mouse model</em></p>
<p>Tengeler AC <em>et
al.</em> (2020) <a href="https://doi.org/10.1186/s40168-020-00816-x"><strong>Gut microbiota from persons with
attention-deficit/hyperactivity disorder affects the brain in
mice</strong></a>. Microbiome
8:44.</p>
<p>In this study, mice are colonized with microbiota from participants
with ADHD (attention deficit hyperactivity disorder) and healthy
participants. The aim of the study was to assess whether the mice
display ADHD behaviors after being inoculated with ADHD microbiota,
suggesting a role of the microbiome in ADHD pathology.</p>
<p>Download the data from
<a href="https://github.com/microbiome/course_2021_radboud/tree/main/data">data</a>
subfolder.</p>
<p><strong>Option 2</strong></p>
<p><em>Western diet intervention study in serotonin transporter modified mouse model</em></p>
<p>Mice received three weeks of Western (high fat, high sugar) or Control diet in
three Genotype groups, a wild-type group and 2 groups with either partial or
complete knock-out of the serotonin transporter (SERT), leading to an excess of
serotonin in the gut, brain and blood circulation. Mice were measured before
and after the intervention. In total there are three factors in this design,
Time (pre-,post-intervention), Diet (Control, Western diet) and Genotype group
(WT, HET, KO). There are about 6-8 mice per group.</p>
<p>Data can be found from Summer School’s <a href="https://brightspace.ru.nl/d2l/le/content/249077/Home">Brightspace</a>.</p>
<p><strong>Option 3</strong></p>
<p><em>Open data set of your own choice</em>, see e.g.:</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/data/experiment/html/microbiomeDataSets.html">Bioconductor microbiomeDataSets</a></li>
</ul>
</div>
<div id="importing-microbiome-data-in-r" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Importing microbiome data in R</h2>
<p><strong>Import example data</strong> by modifying the examples in the online book
section on <a href="https://microbiome.github.io/OMA/data-introduction.html#loading-experimental-microbiome-data">data exploration and
manipulation</a>. The
data files in our example are in <em>biom</em> format, which is a standard
file format for microbiome data. Other file formats exist as well, and
import details vary by platform.</p>
<p>Here, we import <em>biom</em> data files into a specific data container (structure)
in R, <em>TreeSummarizedExperiment</em> (TSE) <a href="https://f1000research.com/articles/9-1246">Huang et
al. (2020)</a>. This provides
the basis for downstream data analysis in the <em>miaverse</em> data science
framework.</p>
<p>In this course, we focus on downstream analysis of taxonomic profiling
data, and assume that the data has already been appropriately
preprocessed and available in the TSE format. In addition to our
example data, further demonstration data sets are readily available in
the TSE format through
<a href="https://bioconductor.org/packages/release/data/experiment/html/microbiomeDataSets.html">microbiomeDataSets</a>.</p>
<p><img src="https://raw.githubusercontent.com/FelixErnst/TreeSummarizedExperiment
/2293440c6e70ae4d6e978b6fdf2c42fdea7fb36a/vignettes/tse2.png" width="100%"/></p>
<p><strong>Figure sources:</strong></p>
<p><strong>Original article</strong>
- Huang R <em>et al</em>. (2021) <a href="https://doi.org/10.12688/%20f1000research.26669.2">TreeSummarizedExperiment: a S4 class
for data with hierarchical structure</a>. F1000Research 9:1246.</p>
<p><strong>Reference Sequence slot extension</strong>
- Lahti L <em>et al</em>. (2020) <a href="https://doi.org/10.7490/%20f1000research.1118447.1">Upgrading the R/Bioconductor ecosystem for microbiome
research</a> F1000Research 9:1464 (slides).</p>
</div>
<div id="example-solutions" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Example solutions</h2>
<ul>
<li>Example code for data import: <a href="import.Rmd">import.Rmd</a></li>
</ul>

<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MatrixGenerics&#39;:
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: SingleCellExperiment</code></pre>
<pre><code>## Loading required package: TreeSummarizedExperiment</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit</code></pre>
<pre><code>## Loading required package: MultiAssayExperiment</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## Loading required package: ggraph</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:Biostrings&#39;:
## 
##     collapse, intersect, setdiff, setequal, union</code></pre>
<pre><code>## The following object is masked from &#39;package:XVector&#39;:
## 
##     slice</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from &#39;package:GenomicRanges&#39;:
## 
##     intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:GenomeInfoDb&#39;:
## 
##     intersect</code></pre>
<pre><code>## The following objects are masked from &#39;package:IRanges&#39;:
## 
##     collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:S4Vectors&#39;:
## 
##     first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:BiocGenerics&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:matrixStats&#39;:
## 
##     count</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="reproducible-reporting-with-rmarkdown.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="microbiome-data-exploration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["radboud2021_material.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
