# Speed comparisons between `tse` and `pseq`containers

Tree summarized experiment objects (tse) and phyloseq (pseq) objects can be viewed as two distinct containers with the same content organized in two different ways. Consequently, these two classes of objects do not perform equally when subject to analogous computational tasks. While the time execution might not diverge significantly in the case of smaller data sets, it could be of matter for larger data sets.

To estimate the time efficiency of homologous tasks but with different targets (`tse` or `pseq`), the following working routines are performed on data sets of various size and compared in terms of the targeted object class:

* Melting
* Transformation
* Agglomeration
* Alpha diversity estimation
* Beta diversity estimation

The analyzed data sets include:

* AsnicarF_2017.relative_abundance
* GlobalPatterns
* VincentC_2016
* SilvermanAGutData
* SongQAData
* SprockettTHData

```{r}
# call packages to load data sets
library(dplyr)
library(mia)
library(microbiomeDataSets)
library(curatedMetagenomicData)
library(ggplot2)
library(scater)
```

```{r}
# list names of data sets
data_sets <- c("AsnicarF_2017.relative_abundance", "GlobalPatterns", "VincentC_2016", "SilvermanAGutData", "SongQAData", "SprockettTHData")
# GrieneisenTSData will be added when conversion to pseq is fixed
```

```{r}
# define matrices for data on dimensions and excution times
cur_it <- 0
tse <- NA
pseq <- NA
dimensions <- matrix(nrow = length(data_sets), ncol = 2)
diff_tse <- matrix(nrow = length(data_sets), ncol = 5)
diff_pseq <- matrix(nrow = length(data_sets), ncol = 5)
assay_values <- ""
```

```{r}
# compare execution times
for (data_set in data_sets) {
  
  cur_it <- cur_it + 1
  
  if (data_set == "GlobalPatterns") {
    
    mapply(data, list = data_set, package = "mia")
    tse <- eval(parse(text = data_set))
    mapply(data, list = data_set, package = "phyloseq")
    pseq <- eval(parse(text = data_set))
    assay_values <- "counts"
      
  } else if (data_set %in% c("SilvermanAGutData", "SongQAData", "SprockettTHData", "GrieneisenTSData")) {
    
    tse <- eval((parse(text = paste0(data_set, "()"))))
    pseq <- makePhyloseqFromTreeSummarizedExperiment(tse)
    assay_values <- "counts"
    
  } else if (data_set %in% c("AsnicarF_2017.relative_abundance", "VincentC_2016")) {
    
    tse <- curatedMetagenomicData(data_set, dryrun = FALSE, counts = TRUE)
    
    if (data_set == "AsnicarF_2017.relative_abundance") {
      tse <- tse[[1]]
    } else if (data_set == "VincentC_2016") {
      tse <- tse[[6]]
    }
    
    assay_values <- "relative_abundance"
    pseq <- makePhyloseqFromTreeSummarizedExperiment(tse,
                                                     abund_values = assay_values)
    
  }
  
  # store dimensions
  dimensions[cur_it, ] <- dim(tse)
  
  # test melting for tse
  start.time1 <- Sys.time()
  molten_tse <- mia::meltAssay(tse,
                       abund_values = assay_values,
                       add_row_data = TRUE,
                       add_col_data = TRUE)
  end.time1 <- Sys.time()
  diff_tse[cur_it, 1] <- end.time1 - start.time1
  
  # test melting for pseq
  start.time2 <- Sys.time()
  molten_pseq <- phyloseq::psmelt(pseq)
  end.time2 <- Sys.time()
  diff_pseq[cur_it, 1] <- end.time2 - start.time2
  
  # test transformation for tse
  start.time1 <- Sys.time()
  trans_tse <- mia::transformSamples(tse,
                                  method = "log10",
                                  pseudocount = 1,
                                  abund_values = assay_values)
  end.time1 <- Sys.time()
  diff_tse[cur_it, 2] <- end.time1 - start.time1
  
  # test transformation for pseq
  start.time2 <- Sys.time()
  trans_pseq <- microbiome::transform(pseq,
                                    transform = "log10p",
                                    target = "sample")
  end.time2 <- Sys.time()
  diff_pseq[cur_it, 2] <- end.time2 - start.time2
  
  # test agglomeration for tse
  start.time1 <- Sys.time()
  tse_phylum <- agglomerateByRank(tse,
                               rank = "Phylum",
                               na.rm = TRUE)
  end.time1 <- Sys.time()
  diff_tse[cur_it, 3] <- end.time1 - start.time1
  
  # test agglomeration for pseq
  start.time2 <- Sys.time()
  pseq_phylum <- phyloseq::tax_glom(pseq,
                                  taxrank = "Phylum")
  # na.rm = TRUE by default in tax_glom
  end.time2 <- Sys.time()
  diff_pseq[cur_it, 3] <- end.time2 - start.time2
  
  # test alpha diversity estimation for tse
  start.time1 <- Sys.time()
  alpha_tse <- mia::estimateDiversity(tse, 
                             abund_values = assay_values,
                             index = "shannon", 
                             name = "shannon")
  end.time1 <- Sys.time()
  diff_tse[cur_it, 4] <- end.time1 - start.time1
  
  # test alpha diversity estimation for pseq
  start.time2 <- Sys.time()
  alpha_pseq <- microbiome::diversity(pseq,
                     index = "shannon")
  end.time2 <- Sys.time()
  diff_pseq[cur_it, 4] <- end.time2 - start.time2
  
  if (data_set != "SilvermanAGutData") {
    
    # test beta diversity estimation for tse
    start.time1 <- Sys.time()
    beta_tse <- runMDS(tse, FUN = vegan::vegdist, name = "MDS_BC", exprs_values = assay_values)
    plotReducedDim(beta_tse, "MDS_BC")
    end.time1 <- Sys.time()
    diff_tse[cur_it, 5] <- end.time1 - start.time1
  
    # test beta diversity estimation for pseq
    start.time2 <- Sys.time()
    beta_pseq <- phyloseq::ordinate(pseq, "MDS", "bray")
    plot_ordination(pseq, beta_pseq, type = "samples")
    end.time2 <- Sys.time()
    diff_pseq[cur_it, 5] <- end.time2 - start.time2
    
  }
  
}
```

```{r}
# view results
dimensions
diff_tse
diff_pseq
```

```{r}
# create architecture for data frame
raw_matrix <- cbind(cbind(rep(dimensions[ , 1], 2), rep(dimensions[ , 2], 2), rbind(diff_tse, diff_pseq)))

object_types <- c(rep("tse", length(data_sets)), rep("pseq", length(data_sets)))

melt_command <- c(rep("mia_meltAssay", length(data_sets)), rep("phyloseq_psMelt", length(data_sets)))

transform_command <- c(rep("mia_transformSamples", length(data_sets)), rep("microbiome_transform", length(data_sets)))

agglomerate_command <- c(rep("mia_agglomerateByRank", length(data_sets)), rep("phyloseq_tax_glom", length(data_sets)))

alpha_command <- c(rep("mia_estimateDiversity", length(data_sets)), rep("microbiome_diversity", length(data_sets)))

beta_command <- c(rep("runMDS", length(data_sets)), rep("ordinate", length(data_sets)))
```

```{r}
# create data frame
execution_times <- as.data.frame(raw_matrix) %>% rename(Features = V1, Samples = V2, Melt = V3, Transform = V4, Agglomerate = V5, AlphaEstimation = V6, BetaEstimation = V7) %>% mutate(ObjectType = object_types, MeltCommand = melt_command, TransformCommand = transform_command, AgglomerateCommand = agglomerate_command, AlphaCommand = alpha_command, BetaCommand = beta_command)
```

### Execution time of melting

```{r}
# compare melting in terms of samples
ggplot(execution_times, aes(x = Samples, y = Melt, color = MeltCommand, shape = as.factor(Features))) + 
  geom_point() +
  labs(title = "Melting comparison", x = "Number of samples", y = "Execution time in s", color = "method:", shape = "number of features:", caption = "Execution time of melting as a function of number of samples") +
  theme_classic() +
  theme(legend.position = "bottom")
```

```{r}
# compare melting in terms of features
ggplot(execution_times, aes(x = Features, y = Melt, color = MeltCommand, shape = as.factor(Samples))) + 
  geom_point() +
  labs(title = "Melting comparison", x = "Number of features", y = "Execution time in s", color = "method:", shape = "number of samples:", caption = "Execution time of melting as a function of number of features") +
  theme_classic() +
  theme(legend.position = "bottom")
```

### Execution time of transformation

```{r}
# compare transformation in terms of samples
ggplot(execution_times, aes(x = Samples, y = Transform, color = TransformCommand, shape = as.factor(Features))) + 
  geom_point() +
  labs(title = "Transformation comparison", x = "Number of samples", y = "Execution time in s", color = "method:", shape = "number of features:", caption = "Execution time of transformation as a function of number of samples") +
  theme_classic() +
  theme(legend.position = "bottom")
```

```{r}
# compare transformation in terms of features
ggplot(execution_times, aes(x = Features, y = Transform, color = TransformCommand, shape = as.factor(Samples))) + 
  geom_point() +
  labs(title = "Transformation comparison", x = "Number of features", y = "Execution time in s", color = "method:", shape = "number of samples:", caption = "Execution time of transformation as a function of number of features") +
  theme_classic() +
  theme(legend.position = "bottom")
```

### Execution time of agglomeration

```{r}
# compare agglomeration in terms of samples
ggplot(execution_times, aes(x = Samples, y = Agglomerate, color = AgglomerateCommand, shape = as.factor(Features))) + 
  geom_point() +
  labs(title = "Agglomeration comparison", x = "Number of samples", y = "Execution time in s", color = "method:", shape = "number of features:", caption = "Execution time of agglomeration as a function of number of samples") +
  theme_classic() +
  theme(legend.position = "bottom")
```

```{r}
# compare agglomeration in terms of features
ggplot(execution_times, aes(x = Features, y = Agglomerate, color = AgglomerateCommand, shape = as.factor(Samples))) + 
  geom_point() +
  labs(title = "Agglomeration comparison", x = "Number of features", y = "Execution time in s", color = "method:", shape = "number of samples:", caption = "Execution time of agglomeration as a function of number of features") +
  theme_classic() +
  theme(legend.position = "bottom")
```

### Execution time of alpha diversity estimation

```{r}
# compare estimation of alpha diversity in terms of samples
ggplot(execution_times, aes(x = Samples, y = AlphaEstimation, color = AlphaCommand, shape = as.factor(Features))) + 
  geom_point() +
  labs(title = "Alpha Diversity Estimation comparison", x = "Number of samples", y = "Execution time in s", color = "method:", shape = "number of features:", caption = "Execution time of estimating alpha diversity as a function of number of samples") +
  theme_classic() +
  theme(legend.position = "bottom")
```

```{r}
# compare estimation of alpha diversity in terms of features
ggplot(execution_times, aes(x = Features, y = AlphaEstimation, color = AlphaCommand, shape = as.factor(Samples))) + 
  geom_point() +
  labs(title = "Alpha Diversity Estimation comparison", x = "Number of features", y = "Execution time in s", color = "method:", shape = "number of samples:", caption = "Execution time of estimating alpha diversity as a function of number of features") +
  theme_classic() +
  theme(legend.position = "bottom")
```

### Execution time of beta diversity estimation

```{r}
# compare estimation of beta diversity in terms of samples
ggplot(execution_times, aes(x = Samples, y = BetaEstimation, color = BetaCommand, shape = as.factor(Features))) + 
  geom_point() +
  labs(title = "Beta Diversity Estimation comparison", x = "Number of samples", y = "Execution time in s", color = "method:", shape = "number of features:", caption = "Execution time of estimating beta diversity as a function of number of samples") +
  theme_classic() +
  theme(legend.position = "bottom")
```

```{r}
# compare estimation of beta diversity in terms of features
ggplot(execution_times, aes(x = Features, y = BetaEstimation, color = BetaCommand, shape = as.factor(Samples))) + 
  geom_point() +
  labs(title = "Beta Diversity Estimation comparison", x = "Number of features", y = "Execution time in s", color = "method:", shape = "number of samples:", caption = "Execution time of estimating beta diversity as a function of number of features") +
  theme_classic() +
  theme(legend.position = "bottom")
```

```{r}
# summarize mean difference in time
mean_time <- execution_times %>% group_by(ObjectType) %>% summarize(mean_melt = mean(Melt), mean_transform = mean(Transform), mean_agglomerate = mean(Agglomerate), mean_alpha = mean(AlphaEstimation), mean_beta = mean(BetaEstimation, na.rm = TRUE))
mean_time
```
