
<!DOCTYPE HTML>
<html lang="" >
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>Differential abundance analysis Â· GitBook</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="description" content="">
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-highlight/website.css">
                
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-search/search.css">
                
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    
        
    
        
    
        
    
        
    
        
    
        
    

        
    
    
    <meta name="HandheldFriendly" content="true"/>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="gitbook/images/favicon.ico" type="image/x-icon">

    
    
    <link rel="prev" href="dmm.html" />
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary">
        
            
<div id="book-search-input" role="search">
    <input type="text" placeholder="Type to search" />
</div>

            
                <nav role="navigation">
                


<ul class="summary">
    
    

    

    
        
        
    
        <li class="chapter " data-level="1.1" data-path="./">
            
                <a href="./">
            
                    
                    Introduction
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2" data-path="introduction.html">
            
                <a href="introduction.html">
            
                    
                    Introduction
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.3" data-path="install.html">
            
                <a href="install.html">
            
                    
                    Install necessary packages
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4" data-path="import.html">
            
                <a href="import.html">
            
                    
                    Import data
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5" data-path="explore.html">
            
                <a href="explore.html">
            
                    
                    Explore data
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.6" data-path="alpha.html">
            
                <a href="alpha.html">
            
                    
                    Calculate alpha diversity
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7" data-path="beta.html">
            
                <a href="beta.html">
            
                    
                    Calculate beta diversity
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.8" data-path="dmm.html">
            
                <a href="dmm.html">
            
                    
                    DMM analysis
            
                </a>
            

            
        </li>
    
        <li class="chapter active" data-level="1.9" data-path="abundance.html">
            
                <a href="abundance.html">
            
                    
                    Differential abundance analysis
            
                </a>
            

            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            Published with GitBook
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href="." >Differential abundance analysis</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
<div id="book-search-results">
    <div class="search-noresults">
    
                                <section class="normal markdown-section">
                                
                                <h1 id="differential-abundance-analysis">Differential abundance analysis</h1>
<p>In this notebook, we analyse abundances with two different methods:
<strong>Wilcoxon test</strong>, and <strong>DESeq2</strong>. Both of these test statistical
differences between groups. We analyse Genus level abundances.</p>
<h3 id="wilcoxon-test">Wilcoxon test</h3>
<p>Wilcoxon test estimates difference between two groups. It is a
non-parametric alternative to t-test, which means that Wilcoxon test
does not require normally distributed data.</p>
<p>Let&#x2019;s first collect the data for the testing purpose.</p>
<pre><code># Agglomerates data to Genus level
tse_genus &lt;- agglomerateByRank(tse, rank = &quot;Genus&quot;)

# Does clr transformation. Pseudocount is added, because data contains zeros, and
# clr transformation includes log transformation.
tse_genus &lt;- transformCounts(tse_genus, method = &quot;clr&quot;, pseudocount = 1)

# Does transpose, so samples are in rows, then creates a data frame.
abundance_analysis_data &lt;- data.frame(t(assay(tse_genus, &quot;clr&quot;)))

# Then we need variable for grouping samples. &quot;patient_status&quot; column includes information
# about patients&apos; status. There two groups &quot;ADHD&quot; and &quot;control&quot;. Let&apos;s include that to the data frame.
abundance_analysis_data &lt;- cbind(abundance_analysis_data, patient_status = colData(tse_genus)$patient_status)
</code></pre><p>Now we can do the Wilcoxon test. We test all the taxa by looping through
columns, and store individual p-values to a vector. Then we create a
data frame from collected data.</p>
<p>Does Wilcoxon test only for columns that contain abundances, not for
column that contain patient status.</p>
<pre><code>colnames &lt;- names(abundance_analysis_data[, !names(abundance_analysis_data) %in% &quot;patient_status&quot;])

wilcoxon_p &lt;- c() # Initialize empty vector for p-values

# Do for loop over selected column names
for (i in colnames) {

  result &lt;- wilcox.test(abundance_analysis_data[, i] ~ patient_status,
                        data = abundance_analysis_data)

  # Stores p-value to the vector with this column name
  wilcoxon_p[[i]]  &lt;- result$p.value

}

wilcoxon_p &lt;- data.frame(taxa =  names(wilcoxon_p),
                         p_raw = unlist(wilcoxon_p))
</code></pre><p>Multiple tests were performed. These are not independent, so we need to
adjust p-values for multiple testing. Otherwise, we would increase the
chance of a type I error drastically depending on our p-value threshold.
By applying p-value adjustment, we can keep the false positive rate at a
level that is acceptable. What is acceptable depends on our research
goals. Here we use the fdr method, but there are several other methods
as well.</p>
<pre><code>wilcoxon_p$p_adjusted &lt;- p.adjust(wilcoxon_p$p_raw, method = &quot;fdr&quot;)

df &lt;- data.frame(x = c(wilcoxon_p$p_raw, wilcoxon_p$p_adjusted), 
                type=rep(c(&quot;raw&quot;, &quot;holm&quot;),
        c(length(wilcoxon_p$p_raw),
          length(wilcoxon_p$p_adjusted))))

wilcoxon_plot &lt;- ggplot(df) +
  geom_histogram(aes(x=x, fill=type)) +
  labs(x = &quot;p-value&quot;, y = &quot;Frequency&quot;) 

wilcoxon_plot

## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
</code></pre><p><img src="abundance_files/figure-markdown_strict/unnamed-chunk-4-1.png" alt=""></p>
<h3 id="deseq2">DESeq2</h3>
<p>Our second analysis method is DESeq2. This performs the data
normalization automatically. It also takes care of the p-value
adjustment, so we don&#x2019;t have to worry about that.</p>
<p>DESeq2 uses negative binomial distribution to detect differences in read
counts between groups. Its normalization takes care of the differences
between library sizes and compositions. DESeq2 analysis includes
multiple steps, but they are done automatically. More information can be
found, e.g., from Harvard Chan Bioinformatic Core&#x2019;s tutorial
<a href="https://hbctraining.github.io/DGE_workshop/lessons/04_DGE_DESeq2_analysis.html" target="_blank">Introduction to DGE -
ARCHIVED</a></p>
<p>Now let us show how to do this. Start by loading the libraries.</p>
<pre><code>library(DESeq2)
library(dplyr)
</code></pre><p>Run the DESeq2 analysis</p>
<pre><code># Creates DESeq2 object from the data. Uses &quot;patient_status&quot; to create groups. 
ds2 &lt;- DESeqDataSet(tse_genus, ~patient_status)

## converting counts to integer mode

## Warning in DESeqDataSet(tse_genus, ~patient_status): 2 duplicate rownames were renamed by adding numbers

## Warning in DESeqDataSet(tse_genus, ~patient_status): some variables in design formula are characters, converting to factors

# Does the analysis
dds &lt;- DESeq(ds2)

## estimating size factors

## estimating dispersions

## gene-wise dispersion estimates

## mean-dispersion relationship

## final dispersion estimates

## fitting model and testing

## -- replacing outliers and refitting for 11 genes
## -- DESeq argument &apos;minReplicatesForReplace&apos; = 7 
## -- original counts are preserved in counts(dds)

## estimating dispersions

## fitting model and testing

# Gets the results from the object
res &lt;- results(dds)

# Creates a data frame from results
df &lt;- as.data.frame(res)

# Adds taxon column that includes names of taxa
df$taxon &lt;- rownames(df)

# Orders the rows of data frame in increasing order firstly based on column &quot;log2FoldChange&quot; 
# and secondly based on &quot;padj&quot; column
df &lt;- df %&gt;% arrange(log2FoldChange, padj)

print(head(knitr::kable((df))))

## [1] &quot;|                                            |     baseMean| log2FoldChange|     lfcSE|        stat|    pvalue|      padj|taxon                                       |&quot;
## [2] &quot;|:-------------------------------------------|------------:|--------------:|---------:|-----------:|---------:|---------:|:-------------------------------------------|&quot;
## [3] &quot;|Genus:Ruminococcaceae_UCG-014               |   22.5482967|    -24.8912675| 2.4606839| -10.1155893| 0.0000000| 0.0000000|Genus:Ruminococcaceae_UCG-014               |&quot;
## [4] &quot;|Order:Bacteroidales                         |   40.3537330|     -9.2417977| 2.1362046|  -4.3262698| 0.0000152| 0.0002730|Order:Bacteroidales                         |&quot;
## [5] &quot;|Genus:Faecalibacterium                      |  231.0795021|     -7.0744335| 1.7456124|  -4.0526944| 0.0000506| 0.0006835|Genus:Faecalibacterium                      |&quot;
## [6] &quot;|Genus:Catabacter                            |   18.0456142|     -6.6154539| 1.7161497|  -3.8548233| 0.0001158| 0.0012508|Genus:Catabacter                            |&quot;
</code></pre><h3 id="ancom-bc">ANCOM-BC</h3>
<p><a href="https://www.nature.com/articles/s41467-020-17041-7" target="_blank">The analysis of composition of microbiomes with bias correction
(ANCOM-BC)</a> is a
recently developed method for differential abundance testing. It is
based on an <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4450248/" target="_blank">earlier published
approach</a>. This
method could be recommended as part of several approaches: A <a href="https://www.biorxiv.org/content/10.1101/2021.05.10.443486v1.full" target="_blank">recent
study</a>
compared several mainstream methods and found that among another method,
ANCOM-BC produced the most consistent results and is probably a
conservative approach. Please note that based on this and other
comparisons, no single method can be recommended across all datasets.
Rather, it could be recommended to apply several methods and look at the
overlap/differences.</p>
<p>As the only method, ANCOM-BC incorporates the so called <em>sampling
fraction</em> into the model. The latter term could be empirically estimated
by the ratio of the library size to the microbial load. Variations in
this sampling fraction would bias differential abundance analyses if
ignored. Furthermore, this method provides p-values, and confidence
intervals for each taxon. It also controls the FDR and it is
computationally simple to implement.</p>
<p>As we will see below, to obtain results, all that is needed is to pass a
phyloseq object to the <code>ancombc()</code> function. Therefore, below we first
convert our <code>tse</code> object to a <code>phyloseq</code> object. Then, we specify the
formula. In this formula, other covariates could potentially be included
to adjust for confounding. Please check the <a href="https://rdrr.io/github/FrederickHuangLin/ANCOMBC/man/ancombc.html" target="_blank">function
documentation</a>
to learn about the additional arguments that we specify below.</p>
<pre><code>library(ANCOMBC)

# currently, ancombc requires the phyloseq format, but we can easily convert:
pseq &lt;- makePhyloseqFromTreeSummarizedExperiment(tse)
pseq_genus &lt;- phyloseq::tax_glom(pseq, taxrank = &quot;Genus&quot;)

out = ancombc(
  phyloseq = pseq_genus, 
  formula = &quot;patient_status&quot;, 
  p_adj_method = &quot;holm&quot;, 
  zero_cut = 0.90, 
  lib_cut = 0, 
  group = &quot;patient_status&quot;, 
  struc_zero = TRUE, 
  neg_lb = TRUE, 
  tol = 1e-5, 
  max_iter = 100, 
  conserve = TRUE, 
  alpha = 0.05, 
  global = TRUE
)
res &lt;- out$res
</code></pre><p>The object <code>out</code> contains all relevant information. Again, see the
<a href="https://rdrr.io/github/FrederickHuangLin/ANCOMBC/man/ancombc.html" target="_blank">documentation of the
function</a>
under <strong>Value</strong> for an explanation of all the output objects. Our
question can be answered by looking at the <code>res</code> object, which now
contains dataframes with the coefficients, standard errors, p-values and
q-values. Conveniently, there is a dataframe <code>diff_abn</code>. Here, we can
find all differentiallt abundant taxa. Below we show the first 6 entries
of this dataframe:</p>
<pre><code>knitr::kable(head(res$diff_abn))
</code></pre><table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">patient_statusControl</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">172647198</td>
<td style="text-align: left;">FALSE</td>
</tr>
<tr class="even">
<td style="text-align: left;">1726478</td>
<td style="text-align: left;">FALSE</td>
</tr>
<tr class="odd">
<td style="text-align: left;">172647201</td>
<td style="text-align: left;">FALSE</td>
</tr>
<tr class="even">
<td style="text-align: left;">17264798</td>
<td style="text-align: left;">FALSE</td>
</tr>
<tr class="odd">
<td style="text-align: left;">172647195</td>
<td style="text-align: left;">FALSE</td>
</tr>
<tr class="even">
<td style="text-align: left;">1726472</td>
<td style="text-align: left;">FALSE</td>
</tr>
</tbody>
</table>

<p>In total, this method detects 13 differentially abundant taxa.</p>
<h3 id="comparison-of-wilcoxon-test-and-deseq2">Comparison of Wilcoxon test and DESeq2</h3>
<p>Let&#x2019;s compare results that we got from the Wilcoxon test and DESeq2. As
we can see from the scatter plot, DESeq2 gives lower p-values than
Wilcoxon test.</p>
<pre><code>mf &lt;- data.frame(df$padj, wilcoxon_p$p_adjusted)
p &lt;- ggplot(mf, aes(x = df$padj, y = wilcoxon_p$p_adjusted)) +
       labs(x = &quot;DESeq2 adjusted p-value&quot;, y = &quot;t-test adjusted p-value&quot;) +
       geom_count() +
 scale_size_area(max_size = 10)

print(p)
</code></pre><p><img src="abundance_files/figure-markdown_strict/unnamed-chunk-9-1.png" alt=""></p>
<p>Prints number of p-values under 0.05</p>
<pre><code>print(paste0(&quot;DESeq2 p-values under 0.05: &quot;, sum(df$padj&lt;0.05, na.rm = TRUE), &quot;/&quot;, length(df$padj)))

## [1] &quot;DESeq2 p-values under 0.05: 7/54&quot;

print(paste0(&quot;Wilcoxon test p-values under 0.05: &quot;, sum(wilcoxon_p$p_adjusted&lt;0.05, na.rm = TRUE), &quot;/&quot;, length(wilcoxon_p$p_adjusted)))

## [1] &quot;Wilcoxon test p-values under 0.05: 2/54&quot;
</code></pre><h3 id="comparison-of-abundance">Comparison of abundance</h3>
<p>In previous steps, we got information which taxa vary between ADHD and
control groups. Let&#x2019;s plot those taxa in the boxplot, and compare
visually if abundances of those taxa differ in ADHD and control samples.
For comparison, let&#x2019;s plot also taxa that do not differ between ADHD and
control groups.</p>
<pre><code># There are some taxa that do not include Genus level information. They are
# excluded from analysis.
# str_detect finds if the pattern is present in values of &quot;taxon&quot; column.
# Subset is taken, only those rows are included that do not include the pattern.
df &lt;- df[ !stringr::str_detect(df$taxon, &quot;Genus:uncultured&quot;), ]

# Sorts p-values in increasing order. Takes 3rd first ones. Takes those rows that match
# with p-values. Takes taxa. 
highest3 &lt;- df[df$padj %in% sort(df$padj, decreasing = FALSE)[1:3], ]$taxon

# From clr transformed table, takes only those taxa that had highest p-values
highest3 &lt;- assay(tse_genus, &quot;clr&quot;)[highest3, ]

# Transposes the table
highest3 &lt;- t(highest3)

# Adds colData that includes patient status infomation
highest3 &lt;- data.frame(highest3, as.data.frame(colData(tse_genus)))

# Some taxa names are that long that they don&apos;t fit nicely into title. So let&apos;s add there 
# a line break after e.g. &quot;Genus&quot;. Here the dot after e.g. Genus is replaced with 
# &quot;: \n&quot;
colnames(highest3)[1:3] &lt;- lapply(colnames(highest3)[1:3], function(x){
  # Replaces the first dot
  temp &lt;- stringr::str_replace(x, &quot;[.]&quot;, &quot;: &quot;)

  # Replace all other dots and underscores with space
  temp &lt;- stringr::str_replace_all(temp, c(&quot;[.]&quot; = &quot; &quot;, &quot;_&quot; = &quot; &quot;))

  # Adds line break so that 30 characters is the maximal width
  temp &lt;- stringr::str_wrap(temp, width = 30)
})

# Sorts p-values in decreasing order. Takes 3rd first ones. Takes those rows that match
# with p-values. Takes taxa. 
lowest3 &lt;- df[df$padj %in% sort(df$padj, decreasing = TRUE)[1:3], ]$taxon

# From clr transformed table, takes only those taxa that had lowest p-values
lowest3 &lt;-assay(tse_genus, &quot;clr&quot;)[lowest3, ]

# Transposes the table
lowest3 &lt;- t(lowest3)

# Adds colData that includes patient status infomation
lowest3 &lt;- data.frame(lowest3, as.data.frame(colData(tse_genus)))

# Some taxa names are that long that they don&apos;t fit nicely into title. So let&apos;s add there 
# a line break after e.g. &quot;Genus&quot;. Here the dot after e.g. Genus is replaced with 
# &quot;: \n&quot;
colnames(lowest3)[1:3] &lt;- lapply(colnames(lowest3)[1:3], function(x){
  # Replaces the first dot
  temp &lt;- stringr::str_replace(x, &quot;[.]&quot;, &quot;: &quot;)

  # Replace all other dots and underscores with space
  temp &lt;- stringr::str_replace_all(temp, c(&quot;[.]&quot; = &quot; &quot;, &quot;_&quot; = &quot; &quot;))

  # Adds line break so that 30 characters is the maximal width
  temp &lt;- stringr::str_wrap(temp, width = 30)
})

# Puts plots in the same picture
gridExtra::grid.arrange(

  # Plot 1
  ggplot(highest3, aes(x = patient_status, y = highest3[,1])) + 
    geom_boxplot() + 
    ylab(&quot;CLR abundances&quot;) + # y axis title
    ggtitle(names(highest3)[1]) + # main title
    theme(title = element_text(size = 7),
          axis.text = element_text(size = 7),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title

  # Plot 2
  ggplot(highest3, aes(x = patient_status, y = highest3[,2])) + 
    geom_boxplot() + 
    ylab(&quot;CLR abundances&quot;) + # y axis title
    ggtitle(names(highest3)[2]) + # main title
    theme(title = element_text(size = 7),
          axis.text = element_text(size = 7),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title

  # Plot 3
  ggplot(highest3, aes(x = patient_status, y = highest3[,3])) + 
    geom_boxplot() + 
    ylab(&quot;CLR abundances&quot;) + # y axis title
    ggtitle(names(highest3)[3]) + # main title
    theme(title = element_text(size = 7),
          axis.text = element_text(size = 7),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title

  # Plot 4
  ggplot(lowest3, aes(x = patient_status, y = lowest3[,1])) + 
    geom_boxplot() + 
    ylab(&quot;CLR abundances&quot;) + # y axis title
    ggtitle(names(lowest3)[1]) + # main title
    theme(title = element_text(size = 7),
          axis.text = element_text(size = 7),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title

  # Plot 5
  ggplot(lowest3, aes(x = patient_status, y = lowest3[,2])) + 
    geom_boxplot() + 
    ylab(&quot;CLR abundances&quot;) + # y axis title
    ggtitle(names(lowest3)[2]) + # main title
    theme(title = element_text(size = 7),
          axis.text = element_text(size = 7),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title

  # Plot 6
  ggplot(lowest3, aes(x = patient_status, y = lowest3[,3])) + 
    geom_boxplot() + 
    ylab(&quot;CLR abundances&quot;) + # y axis title
    ggtitle(names(lowest3)[3]) + # main title
    theme(title = element_text(size = 7),
          axis.text = element_text(size = 7),
          axis.title.x=element_blank()), # makes titles smaller, removes x axis title

  # 3 columns and 2 rows
  ncol = 3, 
  nrow = 2
)
</code></pre><p><img src="abundance_files/figure-markdown_strict/unnamed-chunk-13-1.png" alt=""></p>

                                
                                </section>
                            
    </div>
    <div class="search-results">
        <div class="has-results">
            
            <h1 class="search-results-title"><span class='search-results-count'></span> results matching "<span class='search-query'></span>"</h1>
            <ul class="search-results-list"></ul>
            
        </div>
        <div class="no-results">
            
            <h1 class="search-results-title">No results matching "<span class='search-query'></span>"</h1>
            
        </div>
    </div>
</div>

                        </div>
                    </div>
                
            </div>

            
                
                <a href="dmm.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page: DMM analysis">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"Differential abundance analysis","level":"1.9","depth":1,"previous":{"title":"DMM analysis","level":"1.8","depth":1,"path":"dmm.md","ref":"dmm.md","articles":[]},"dir":"ltr"},"config":{"gitbook":"*","theme":"default","variables":{},"plugins":[],"pluginsConfig":{"highlight":{},"search":{},"lunr":{"maxIndexSize":1000000,"ignoreSpecialCharacters":false},"sharing":{"facebook":true,"twitter":true,"google":false,"weibo":false,"instapaper":false,"vk":false,"all":["facebook","google","twitter","weibo","instapaper"]},"fontsettings":{"theme":"white","family":"sans","size":2},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":false}},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"pdf":{"pageNumbers":true,"fontSize":12,"fontFamily":"Arial","paperSize":"a4","chapterMark":"pagebreak","pageBreaksBefore":"/","margin":{"right":62,"left":62,"top":56,"bottom":56}},"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"}},"file":{"path":"abundance.md","mtime":"2021-06-07T08:58:40.442Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2021-06-07T13:21:45.551Z"},"basePath":".","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="gitbook/gitbook.js"></script>
    <script src="gitbook/theme.js"></script>
    
        
        <script src="gitbook/gitbook-plugin-search/search-engine.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-search/search.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-lunr/lunr.min.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-lunr/search-lunr.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-sharing/buttons.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

