<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Alpha diversity | Miscellaneous material</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Alpha diversity | Miscellaneous material" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/course_2021_radboud" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Alpha diversity | Miscellaneous material" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="Leo Lahti, Tuomas Borman, Henrik Eckerman" />


<meta name="date" content="2021-07-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-exploration.html"/>
<link rel="next" href="beta-diversity.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-miaverse-framework"><i class="fa fa-check"></i><b>1.1</b> The miaverse framework</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#learning-goals"><i class="fa fa-check"></i><b>1.2</b> Learning goals</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="program.html"><a href="program.html"><i class="fa fa-check"></i><b>2</b> Program</a>
<ul>
<li class="chapter" data-level="2.1" data-path="program.html"><a href="program.html#monday-12-july-from-raw-sequences-to-ecological-data-analysis"><i class="fa fa-check"></i><b>2.1</b> Monday 12 July: from raw sequences to ecological data analysis</a></li>
<li class="chapter" data-level="2.2" data-path="program.html"><a href="program.html#tuesday-13-july-key-concepts-in-data-analysis-alpha-diversity"><i class="fa fa-check"></i><b>2.2</b> Tuesday 13 July: Key concepts in data analysis: alpha diversity</a></li>
<li class="chapter" data-level="2.3" data-path="program.html"><a href="program.html#wednesday-14-july-key-concepts-in-data-analysis-beta-diversity"><i class="fa fa-check"></i><b>2.3</b> Wednesday 14 July: Key concepts in data analysis: beta diversity</a></li>
<li class="chapter" data-level="2.4" data-path="program.html"><a href="program.html#thursday-15-july-key-concepts-in-data-analysis-differential-abundance"><i class="fa fa-check"></i><b>2.4</b> Thursday 15 July: Key concepts in data analysis: differential abundance</a></li>
<li class="chapter" data-level="2.5" data-path="program.html"><a href="program.html#friday-16-july-presentations-closing"><i class="fa fa-check"></i><b>2.5</b> Friday 16 July: Presentations &amp; closing</a></li>
<li class="chapter" data-level="2.6" data-path="program.html"><a href="program.html#course-assignments-todo"><i class="fa fa-check"></i><b>2.6</b> Course assignments (TODO)</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="program.html"><a href="program.html#monday-12-july-done"><i class="fa fa-check"></i><b>2.6.1</b> Monday 12 July: DONE</a></li>
<li class="chapter" data-level="2.6.2" data-path="program.html"><a href="program.html#tue-13-july-alpha-diversity"><i class="fa fa-check"></i><b>2.6.2</b> Tue 13 July: alpha diversity</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#checklist-before-the-course"><i class="fa fa-check"></i><b>3.1</b> Checklist (before the course)</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#support-and-resources"><i class="fa fa-check"></i><b>3.2</b> Support and resources</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="install-and-load-required-r-packages.html"><a href="install-and-load-required-r-packages.html"><i class="fa fa-check"></i><b>4</b> Install and load required R packages</a></li>
<li class="chapter" data-level="5" data-path="import-data.html"><a href="import-data.html"><i class="fa fa-check"></i><b>5</b> Import data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="import-data.html"><a href="import-data.html#data"><i class="fa fa-check"></i><b>5.1</b> Data</a></li>
<li class="chapter" data-level="5.2" data-path="import-data.html"><a href="import-data.html#investigate-the-r-data-object"><i class="fa fa-check"></i><b>5.2</b> Investigate the R data object</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="import-data.html"><a href="import-data.html#rowdata-taxonomic-information"><i class="fa fa-check"></i><b>5.2.1</b> rowData (taxonomic information)</a></li>
<li class="chapter" data-level="5.2.2" data-path="import-data.html"><a href="import-data.html#coldata-sample-information"><i class="fa fa-check"></i><b>5.2.2</b> colData (sample information)</a></li>
<li class="chapter" data-level="5.2.3" data-path="import-data.html"><a href="import-data.html#phylogenetic-tree-information"><i class="fa fa-check"></i><b>5.2.3</b> Phylogenetic tree information</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-exploration.html"><a href="data-exploration.html"><i class="fa fa-check"></i><b>6</b> Data exploration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-exploration.html"><a href="data-exploration.html#investigate-the-contents-of-a-microbiome-data-object"><i class="fa fa-check"></i><b>6.1</b> Investigate the contents of a microbiome data object</a></li>
<li class="chapter" data-level="6.2" data-path="data-exploration.html"><a href="data-exploration.html#manipulate-the-data"><i class="fa fa-check"></i><b>6.2</b> Manipulate the data</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="data-exploration.html"><a href="data-exploration.html#transformations"><i class="fa fa-check"></i><b>6.2.1</b> Transformations</a></li>
<li class="chapter" data-level="6.2.2" data-path="data-exploration.html"><a href="data-exploration.html#aggregation"><i class="fa fa-check"></i><b>6.2.2</b> Aggregation</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="data-exploration.html"><a href="data-exploration.html#visualization"><i class="fa fa-check"></i><b>6.3</b> Visualization</a></li>
<li class="chapter" data-level="6.4" data-path="data-exploration.html"><a href="data-exploration.html#further-resources"><i class="fa fa-check"></i><b>6.4</b> Further resources</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>7</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#visualization-1"><i class="fa fa-check"></i><b>7.1</b> Visualization</a></li>
<li class="chapter" data-level="7.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#statistical-testing-and-comparisons"><i class="fa fa-check"></i><b>7.2</b> Statistical testing and comparisons</a></li>
<li class="chapter" data-level="7.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#further-resources-1"><i class="fa fa-check"></i><b>7.3</b> Further resources</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#examples-of-pcoa-with-different-settings"><i class="fa fa-check"></i><b>8.1</b> Examples of PCoA with different settings</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-bray-curtis"><i class="fa fa-check"></i><b>8.1.1</b> PCoA for ASV-level data with Bray-Curtis</a></li>
<li class="chapter" data-level="8.1.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-aitchison-distance"><i class="fa fa-check"></i><b>8.1.2</b> PCoA for ASV-level data with Aitchison distance</a></li>
<li class="chapter" data-level="8.1.3" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-aggregated-to-phylum-level"><i class="fa fa-check"></i><b>8.1.3</b> PCoA aggregated to Phylum level</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#highlighting-external-variables-on-pcoa-plot"><i class="fa fa-check"></i><b>8.2</b> Highlighting external variables on PCoA plot</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-with-discrete-sample-grouping-variable-shown-with-colors"><i class="fa fa-check"></i><b>8.2.1</b> PCoA with discrete sample grouping variable shown with colors</a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-with-continuous-sample-grouping-variable-shown-with-colors"><i class="fa fa-check"></i><b>8.2.2</b> PCoA with continuous sample grouping variable shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#estimating-associations-with-an-external-variable"><i class="fa fa-check"></i><b>8.3</b> Estimating associations with an external variable</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#further-resources-2"><i class="fa fa-check"></i><b>8.4</b> Further resources</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="dirichlet-multinomial-mixture-model.html"><a href="dirichlet-multinomial-mixture-model.html"><i class="fa fa-check"></i><b>9</b> Dirichlet-Multinomial Mixture Model</a>
<ul>
<li class="chapter" data-level="9.1" data-path="dirichlet-multinomial-mixture-model.html"><a href="dirichlet-multinomial-mixture-model.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>9.1</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html"><i class="fa fa-check"></i><b>10</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#wilcoxon-test"><i class="fa fa-check"></i><b>10.1</b> Wilcoxon test</a></li>
<li class="chapter" data-level="10.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#deseq2"><i class="fa fa-check"></i><b>10.2</b> DESeq2</a></li>
<li class="chapter" data-level="10.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#ancom-bc"><i class="fa fa-check"></i><b>10.3</b> ANCOM-BC</a></li>
<li class="chapter" data-level="10.4" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-wilcoxon-test-and-deseq2"><i class="fa fa-check"></i><b>10.4</b> Comparison of Wilcoxon test and DESeq2</a></li>
<li class="chapter" data-level="10.5" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-abundance"><i class="fa fa-check"></i><b>10.5</b> Comparison of abundance</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="study-material.html"><a href="study-material.html"><i class="fa fa-check"></i><b>11</b> Study material</a>
<ul>
<li class="chapter" data-level="11.1" data-path="study-material.html"><a href="study-material.html#lecture-slides"><i class="fa fa-check"></i><b>11.1</b> Lecture slides</a></li>
<li class="chapter" data-level="11.2" data-path="study-material.html"><a href="study-material.html#example-solutions"><i class="fa fa-check"></i><b>11.2</b> Example solutions</a></li>
<li class="chapter" data-level="11.3" data-path="study-material.html"><a href="study-material.html#r-programming-resources"><i class="fa fa-check"></i><b>11.3</b> R programming resources</a></li>
<li class="chapter" data-level="11.4" data-path="study-material.html"><a href="study-material.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>11.4</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="11.5" data-path="study-material.html"><a href="study-material.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>11.5</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="11.6" data-path="study-material.html"><a href="study-material.html#further-reading"><i class="fa fa-check"></i><b>11.6</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html"><i class="fa fa-check"></i><b>12</b> Miscellaneous material</a>
<ul>
<li class="chapter" data-level="12.1" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html#shapiro-wilk-test"><i class="fa fa-check"></i><b>12.1</b> Shapiro-Wilk test</a></li>
<li class="chapter" data-level="12.2" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html#deseq-details"><i class="fa fa-check"></i><b>12.2</b> Deseq details</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Miscellaneous material</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="alpha-diversity" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Alpha diversity</h1>
<p>This section demonstrates how alpha diversity indices are calculated.</p>
<p>Alpha diversity is a quantity that measures diversity of taxa in within the
sample. Higher numbers of unique taxa, and more even abundance
distributions within a sample yield larger values for alpha diversity.</p>
<p>Alpha diversity is an important quantity in a microbiome research. The
mia package provides access to a wide variety of alpha diversity
indices. As an example, we show how to calculate Shannon and Faith
diversity indices.</p>
<p>Shannon index reflects how many different taxa there are and how
evenly they are distributed within a sample. Faith index additionally
takes into account the phylogenetic relations between the taxa when
quantifying the overall community diversity. In both indices, higher
values represent higher diversity.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="alpha-diversity.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Indices to be calculated. If we don&#39;t specify indices, by default, every index</span></span>
<span id="cb43-2"><a href="alpha-diversity.html#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="co"># is calculated.</span></span>
<span id="cb43-3"><a href="alpha-diversity.html#cb43-3" aria-hidden="true" tabindex="-1"></a>indices <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;shannon&quot;</span>, <span class="st">&quot;faith&quot;</span>)</span>
<span id="cb43-4"><a href="alpha-diversity.html#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="alpha-diversity.html#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Indices are stored in colData (i.e., sample metadata). We can specify the name</span></span>
<span id="cb43-6"><a href="alpha-diversity.html#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="co"># of column, or we can use the default name which is the name of index </span></span>
<span id="cb43-7"><a href="alpha-diversity.html#cb43-7" aria-hidden="true" tabindex="-1"></a><span class="co"># (i.e., &quot;shannon&quot; and &quot;faith&quot;). </span></span>
<span id="cb43-8"><a href="alpha-diversity.html#cb43-8" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Shannon_index&quot;</span>, <span class="st">&quot;Faith_diversity_index&quot;</span>)</span>
<span id="cb43-9"><a href="alpha-diversity.html#cb43-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-10"><a href="alpha-diversity.html#cb43-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates indices</span></span>
<span id="cb43-11"><a href="alpha-diversity.html#cb43-11" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">estimateDiversity</span>(tse, <span class="at">index =</span> indices, <span class="at">name =</span> names)</span>
<span id="cb43-12"><a href="alpha-diversity.html#cb43-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-13"><a href="alpha-diversity.html#cb43-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Shows the calculated indices</span></span>
<span id="cb43-14"><a href="alpha-diversity.html#cb43-14" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">colData</span>(tse)[names]))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Shannon_index</th>
<th align="right">Faith_diversity_index</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A110</td>
<td align="right">1.765407</td>
<td align="right">7.39224</td>
</tr>
<tr class="even">
<td align="left">A12</td>
<td align="right">2.716438</td>
<td align="right">6.29378</td>
</tr>
<tr class="odd">
<td align="left">A15</td>
<td align="right">3.178103</td>
<td align="right">6.60608</td>
</tr>
<tr class="even">
<td align="left">A19</td>
<td align="right">2.891987</td>
<td align="right">6.79708</td>
</tr>
<tr class="odd">
<td align="left">A21</td>
<td align="right">2.841979</td>
<td align="right">6.65110</td>
</tr>
<tr class="even">
<td align="left">A23</td>
<td align="right">2.797942</td>
<td align="right">5.96246</td>
</tr>
</tbody>
</table>
<p>Next we can visualize Shannon index with histogram.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="alpha-diversity.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot needs data.frame as input. Because colData is DataFrame, it needs to be </span></span>
<span id="cb44-2"><a href="alpha-diversity.html#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="co"># converted. </span></span>
<span id="cb44-3"><a href="alpha-diversity.html#cb44-3" aria-hidden="true" tabindex="-1"></a>shannon_hist <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)), </span>
<span id="cb44-4"><a href="alpha-diversity.html#cb44-4" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">aes</span>(<span class="at">x =</span> Shannon_index)) <span class="sc">+</span> </span>
<span id="cb44-5"><a href="alpha-diversity.html#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="at">fill =</span> <span class="st">&quot;gray&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb44-6"><a href="alpha-diversity.html#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Shannon index&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Sample frequency&quot;</span>)</span>
<span id="cb44-7"><a href="alpha-diversity.html#cb44-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-8"><a href="alpha-diversity.html#cb44-8" aria-hidden="true" tabindex="-1"></a>shannon_hist</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-30-1.png" width="672" /></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="alpha-diversity.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # Same thing but done differently</span></span>
<span id="cb45-2"><a href="alpha-diversity.html#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="co"># # Creates histogram. With &quot;break&quot;, number of bins can be specified. However, the</span></span>
<span id="cb45-3"><a href="alpha-diversity.html#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co"># # value is taken as a suggestion, because hist() uses pretty() to calculate breakpoints.</span></span>
<span id="cb45-4"><a href="alpha-diversity.html#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co"># hist(colData(tse)$Shannon_index, col = &quot;green&quot;, breaks = 20,</span></span>
<span id="cb45-5"><a href="alpha-diversity.html#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="co"># xlab = &quot;Shannon index&quot;,</span></span>
<span id="cb45-6"><a href="alpha-diversity.html#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ylab = &quot;Sample frequency&quot;,</span></span>
<span id="cb45-7"><a href="alpha-diversity.html#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co"># main = &quot;Histogram of Shannon index&quot;)</span></span></code></pre></div>
<p>To see, if there is dependency between Shannon and Faith, we can do cross-plot i.e.,
scatter plot, where one index is on the x-axis and another on the y-axis.</p>
<p>It seems that, there is a positive correlation between these two indices.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="alpha-diversity.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # Does the same thing but differently</span></span>
<span id="cb46-2"><a href="alpha-diversity.html#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plot(colData(tse)$Shannon_index, colData(tse)$Faith_diversity_index,</span></span>
<span id="cb46-3"><a href="alpha-diversity.html#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="co">#      xlab = &quot;Shannon index&quot;,</span></span>
<span id="cb46-4"><a href="alpha-diversity.html#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co">#      ylab = &quot;Faith diversity index&quot;,</span></span>
<span id="cb46-5"><a href="alpha-diversity.html#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="co">#      main = &quot;plot()&quot;) +</span></span>
<span id="cb46-6"><a href="alpha-diversity.html#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   # Adds regression line</span></span>
<span id="cb46-7"><a href="alpha-diversity.html#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   abline(lm(colData(tse)$Faith_diversity_index ~ colData(tse)$Shannon_index)) </span></span>
<span id="cb46-8"><a href="alpha-diversity.html#cb46-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-9"><a href="alpha-diversity.html#cb46-9" aria-hidden="true" tabindex="-1"></a>cross_plot <span class="ot">&lt;-</span> ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)), </span>
<span id="cb46-10"><a href="alpha-diversity.html#cb46-10" aria-hidden="true" tabindex="-1"></a>                                     <span class="fu">aes</span>(<span class="at">x =</span> Shannon_index, <span class="at">y =</span> Faith_diversity_index)) <span class="sc">+</span> </span>
<span id="cb46-11"><a href="alpha-diversity.html#cb46-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="co"># Adds points</span></span>
<span id="cb46-12"><a href="alpha-diversity.html#cb46-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm) <span class="sc">+</span> <span class="co"># Adds regression line</span></span>
<span id="cb46-13"><a href="alpha-diversity.html#cb46-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Shannon index&quot;</span>) <span class="sc">+</span> <span class="co"># x axis title</span></span>
<span id="cb46-14"><a href="alpha-diversity.html#cb46-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Faith diversity index&quot;</span>)  <span class="co"># y axis title</span></span>
<span id="cb46-15"><a href="alpha-diversity.html#cb46-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-16"><a href="alpha-diversity.html#cb46-16" aria-hidden="true" tabindex="-1"></a>cross_plot</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-32-1.png" width="672" /></p>
<div id="visualization-1" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Visualization</h2>
<p>Next let’s compare indices between different patient status and cohorts. Boxplot is
suitable for that purpose.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="alpha-diversity.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates Shannon boxplot </span></span>
<span id="cb48-2"><a href="alpha-diversity.html#cb48-2" aria-hidden="true" tabindex="-1"></a>shannon_box <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)), <span class="fu">aes</span>(<span class="at">x =</span> patient_status, </span>
<span id="cb48-3"><a href="alpha-diversity.html#cb48-3" aria-hidden="true" tabindex="-1"></a>                                                       <span class="at">y =</span> Shannon_index, <span class="at">fill =</span> cohort)) <span class="sc">+</span> </span>
<span id="cb48-4"><a href="alpha-diversity.html#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb48-5"><a href="alpha-diversity.html#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb48-6"><a href="alpha-diversity.html#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="alpha-diversity.html#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates Faith boxplot </span></span>
<span id="cb48-8"><a href="alpha-diversity.html#cb48-8" aria-hidden="true" tabindex="-1"></a>faith_box <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tse)), <span class="fu">aes</span>(<span class="at">x =</span> patient_status, </span>
<span id="cb48-9"><a href="alpha-diversity.html#cb48-9" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">y =</span> Faith_diversity_index, </span>
<span id="cb48-10"><a href="alpha-diversity.html#cb48-10" aria-hidden="true" tabindex="-1"></a>                                                     <span class="at">fill =</span> cohort)) <span class="sc">+</span> </span>
<span id="cb48-11"><a href="alpha-diversity.html#cb48-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb48-12"><a href="alpha-diversity.html#cb48-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb48-13"><a href="alpha-diversity.html#cb48-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-14"><a href="alpha-diversity.html#cb48-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Puts them into same picture</span></span>
<span id="cb48-15"><a href="alpha-diversity.html#cb48-15" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(shannon_box, faith_box, <span class="at">nrow =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
</div>
<div id="statistical-testing-and-comparisons" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Statistical testing and comparisons</h2>
<p>To further investigate if patient status explains the variation of Shannon index, let’s
do a Wilcoxon test.</p>
<p>The Wilcoxon test is similar to a Student’s t-test, however, the Student’s t-test is parametric.
It means that the data must be normally distributed. The Wilcoxon test is non-parametric
so it doesn’t make any assumptions about the distribution.</p>
<p>The Wilcoxon test tests if there are statistical differences between two groups.
Here it tests, if ADHD and control groups have different Shannon index values.
As we can see, there is no difference between groups, because the p-value is over 0.05,
which is often used as a standard cutoff point.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="alpha-diversity.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Wilcoxon test, where Shannon index is the variable that we are comparing. </span></span>
<span id="cb49-2"><a href="alpha-diversity.html#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Patient status - ADHD or control - is the factor that we use for grouping. </span></span>
<span id="cb49-3"><a href="alpha-diversity.html#cb49-3" aria-hidden="true" tabindex="-1"></a>wilcoxon_shannon <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(Shannon_index <span class="sc">~</span> patient_status, <span class="at">data =</span> <span class="fu">colData</span>(tse))</span>
<span id="cb49-4"><a href="alpha-diversity.html#cb49-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-5"><a href="alpha-diversity.html#cb49-5" aria-hidden="true" tabindex="-1"></a>wilcoxon_shannon</span></code></pre></div>
<pre><code>## 
##  Wilcoxon rank sum exact test
## 
## data:  Shannon_index by patient_status
## W = 76, p-value = 0.4879
## alternative hypothesis: true location shift is not equal to 0</code></pre>
<p>Another test that we can make is to test if ADHD samples differs between different
cohorts. From boxplot that we made in previous step, we can see that there might
be statistically significant difference between different cohorts.</p>
<p>Let’s compare Shannon index of ADHD samples between cohort 2 and cohort 3.</p>
<p>As we can see, there is statistically significant difference between the cohorts.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="alpha-diversity.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Takes subset of colData. Takes only ADHD samples</span></span>
<span id="cb51-2"><a href="alpha-diversity.html#cb51-2" aria-hidden="true" tabindex="-1"></a>ADHD_shannon <span class="ot">&lt;-</span> <span class="fu">colData</span>(tse)[ <span class="fu">colData</span>(tse)[, <span class="st">&quot;patient_status&quot;</span>] <span class="sc">==</span> <span class="st">&quot;ADHD&quot;</span> , ]</span>
<span id="cb51-3"><a href="alpha-diversity.html#cb51-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-4"><a href="alpha-diversity.html#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Takes subset of colData. Takes only samples that are in cohort 2 or cohort 3.</span></span>
<span id="cb51-5"><a href="alpha-diversity.html#cb51-5" aria-hidden="true" tabindex="-1"></a>ADHD_shannon <span class="ot">&lt;-</span> ADHD_shannon[ ADHD_shannon[, <span class="st">&quot;cohort&quot;</span>] <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Cohort_2&quot;</span>, <span class="st">&quot;Cohort_3&quot;</span>) , ]</span>
<span id="cb51-6"><a href="alpha-diversity.html#cb51-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-7"><a href="alpha-diversity.html#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Wilcoxon test, where Shannon index is the variable that we are comparing. </span></span>
<span id="cb51-8"><a href="alpha-diversity.html#cb51-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort - 2 or 3 - is the factor that we use for grouping. </span></span>
<span id="cb51-9"><a href="alpha-diversity.html#cb51-9" aria-hidden="true" tabindex="-1"></a>wilcoxon_shannon_ADHD_cohorts <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(Shannon_index <span class="sc">~</span> cohort, <span class="at">data =</span> ADHD_shannon)</span>
<span id="cb51-10"><a href="alpha-diversity.html#cb51-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-11"><a href="alpha-diversity.html#cb51-11" aria-hidden="true" tabindex="-1"></a>wilcoxon_shannon_ADHD_cohorts</span></code></pre></div>
<pre><code>## 
##  Wilcoxon rank sum exact test
## 
## data:  Shannon_index by cohort
## W = 20, p-value = 0.01587
## alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
<div id="further-resources-1" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Further resources</h2>
<p>For more examples, see a dedicated section on alpha diversity in the
<a href="https://microbiome.github.io/OMA/">online book</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-exploration.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="beta-diversity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
