
<!DOCTYPE HTML>
<html lang="" >
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>Calculate alpha diversity Â· GitBook</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="description" content="">
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-highlight/website.css">
                
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-search/search.css">
                
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    
        
    
        
    
        
    
        
    
        
    
        
    

        
    
    
    <meta name="HandheldFriendly" content="true"/>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="gitbook/images/favicon.ico" type="image/x-icon">

    
    <link rel="next" href="beta.html" />
    
    
    <link rel="prev" href="explore.html" />
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary">
        
            
<div id="book-search-input" role="search">
    <input type="text" placeholder="Type to search" />
</div>

            
                <nav role="navigation">
                


<ul class="summary">
    
    

    

    
        
        
    
        <li class="chapter " data-level="1.1" data-path="./">
            
                <a href="./">
            
                    
                    Introduction
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2" data-path="install.html">
            
                <a href="install.html">
            
                    
                    Install and load required R packages
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.3" data-path="import.html">
            
                <a href="import.html">
            
                    
                    Import data
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4" data-path="explore.html">
            
                <a href="explore.html">
            
                    
                    Explore data
            
                </a>
            

            
        </li>
    
        <li class="chapter active" data-level="1.5" data-path="alpha.html">
            
                <a href="alpha.html">
            
                    
                    Calculate alpha diversity
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.6" data-path="beta.html">
            
                <a href="beta.html">
            
                    
                    Calculate beta diversity
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7" data-path="dmm.html">
            
                <a href="dmm.html">
            
                    
                    DMM analysis
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.8" data-path="abundance.html">
            
                <a href="abundance.html">
            
                    
                    Differential abundance analysis
            
                </a>
            

            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            Published with GitBook
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href="." >Calculate alpha diversity</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
<div id="book-search-results">
    <div class="search-noresults">
    
                                <section class="normal markdown-section">
                                
                                <h1 id="alpha-diversity">Alpha diversity</h1>
<p>This section demonstrates how alpha diversity indices are calculated.</p>
<p>Alpha diversity is a quantity that measures diversity of taxa in one
sample. Higher numbers of unique taxa, and more even abundance
distributions within a sample yield larger values for alpha diversity.</p>
<p>Alpha diversity is an important quantity in microbiome research. The mia
package provides access to a wide variety of alpha diversity indices. As
an example, we show how to calculate Shannon and Faith diversity
indices.</p>
<p>Shannon index reflects how many different taxa there are and how evenly
they are distributed within a sample. Faith index additionally takes
into account the phylogenetic relations between the taxa when
quantifying the overall community diversity. In both indices, higher
values represent higher diversity.</p>
<pre><code># Indices to be calculated. If we don&apos;t specify indices, by default, every index
# is calculated.
indices &lt;- c(&quot;shannon&quot;, &quot;faith&quot;)

# Indices are stored in colData (i.e., sample metadata). We can specify the name
# of column, or we can use the default name which is the name of index 
# (i.e., &quot;shannon&quot; and &quot;faith&quot;). 
names &lt;- c(&quot;Shannon_index&quot;, &quot;Faith_diversity_index&quot;)

# Calculates indices
tse &lt;- estimateDiversity(tse, index = indices, name = names)

# Shows the calculated indices
knitr::kable(head(colData(tse)[names]))
</code></pre><table>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">Shannon_index</th>
<th style="text-align: right;">Faith_diversity_index</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">A110</td>
<td style="text-align: right;">1.765407</td>
<td style="text-align: right;">7.39224</td>
</tr>
<tr class="even">
<td style="text-align: left;">A12</td>
<td style="text-align: right;">2.716438</td>
<td style="text-align: right;">6.29378</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A15</td>
<td style="text-align: right;">3.178103</td>
<td style="text-align: right;">6.60608</td>
</tr>
<tr class="even">
<td style="text-align: left;">A19</td>
<td style="text-align: right;">2.891987</td>
<td style="text-align: right;">6.79708</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A21</td>
<td style="text-align: right;">2.841979</td>
<td style="text-align: right;">6.65110</td>
</tr>
<tr class="even">
<td style="text-align: left;">A23</td>
<td style="text-align: right;">2.797942</td>
<td style="text-align: right;">5.96246</td>
</tr>
</tbody>
</table>

<p>Next we can visualize Shannon index with histogram.</p>
<pre><code># ggplot needs data.frame as input. Because colData is DataFrame, it needs to be 
# converted. 
shannon_hist &lt;- ggplot(as.data.frame(colData(tse)), 
                       aes(x = Shannon_index)) + 
  geom_histogram(bins = 20, fill = &quot;gray&quot;, color = &quot;black&quot;) +
  labs(x = &quot;Shannon index&quot;, y = &quot;Sample frequency&quot;)

shannon_hist
</code></pre><p><img src="alpha_files/figure-markdown_strict/unnamed-chunk-2-1.png" alt=""></p>
<pre><code># # Same thing but done differently
# # Creates histogram. With &quot;break&quot;, number of bins can be specified. However, the
# # value is taken as a suggestion, because hist() uses pretty() to calculate breakpoints.
# hist(colData(tse)$Shannon_index, col = &quot;green&quot;, breaks = 20,
# xlab = &quot;Shannon index&quot;,
# ylab = &quot;Sample frequency&quot;,
# main = &quot;Histogram of Shannon index&quot;)
</code></pre><p>To see, if there is dependency between Shannon and Faith, we can do
cross-plot i.e., scatter plot, where one index is on the x-axis and
another on the y-axis.</p>
<p>It seems that, there is a positive correlation between these two
indices.</p>
<pre><code># # Does the same thing but differently
# plot(colData(tse)$Shannon_index, colData(tse)$Faith_diversity_index,
#      xlab = &quot;Shannon index&quot;,
#      ylab = &quot;Faith diversity index&quot;,
#      main = &quot;plot()&quot;) +
#   # Adds regression line
#   abline(lm(colData(tse)$Faith_diversity_index ~ colData(tse)$Shannon_index)) 

cross_plot &lt;- ggplot2::ggplot(as.data.frame(colData(tse)), 
                                     aes(x = Shannon_index, y = Faith_diversity_index)) + 
  geom_point() + # Adds points
  geom_smooth(method=lm) + # Adds regression line
  xlab(&quot;Shannon index&quot;) + # x axis title
  ylab(&quot;Faith diversity index&quot;)  # y axis title

cross_plot

## `geom_smooth()` using formula &apos;y ~ x&apos;
</code></pre><p><img src="alpha_files/figure-markdown_strict/unnamed-chunk-4-1.png" alt=""></p>
<h2 id="visualization">Visualization</h2>
<p>Next let&#x2019;s compare indices between different genotypes and diets.
Boxplot is suitable for that purpose.</p>
<pre><code># Creates Shannon boxplot 
shannon_box &lt;- ggplot(as.data.frame(colData(tse)), aes(x = patient_status, 
                                                       y = Shannon_index, fill = cohort)) + 
  geom_boxplot() +
  theme(title = element_text(size = 12)) # makes titles smaller

# Creates Faith boxplot 
faith_box &lt;- ggplot(as.data.frame(colData(tse)), aes(x = patient_status, 
                                                     y = Faith_diversity_index, 
                                                     fill = cohort)) + 
  geom_boxplot() +
  theme(title = element_text(size = 12)) # makes titles smaller

# Puts them into same picture
gridExtra::grid.arrange(shannon_box, faith_box, nrow = 2)
</code></pre><p><img src="alpha_files/figure-markdown_strict/unnamed-chunk-5-1.png" alt=""></p>
<h2 id="statistical-testing-and-comparisons">Statistical testing and comparisons</h2>
<p>To further investigate if patient status explains the variation of
Shannon index, let&#x2019;s do Wilcoxon test.</p>
<p>Wilcoxon test is similar to Student&#x2019;s t-test, however, Student&#x2019;s t-test
is parametric. It means that the data must be normally distributed.
Wilcoxon test is non-parametric so it doesn&#x2019;t make any assumptions about
the distribution.</p>
<p>Wilcoxon test tests if there are statistical differences between two
groups. Here it tests, if ADHD and control groups have different Shannon
index values. As we can see, there is no difference between groups,
because p-value is over 0.05, which is often used as a standard cutoff
point.</p>
<pre><code># Wilcoxon test, where Shannon index is the variable that we are comparing. 
# Patient status - ADHD or control - is the factor that we use for grouping. 
wilcoxon_shannon &lt;- wilcox.test(Shannon_index ~ patient_status, data = colData(tse))

wilcoxon_shannon

## 
##  Wilcoxon rank sum exact test
## 
## data:  Shannon_index by patient_status
## W = 76, p-value = 0.4879
## alternative hypothesis: true location shift is not equal to 0
</code></pre><p>Another test that we can make is to test if ADHD samples differs between
different cohorts. From boxplot that we made in previous step, we can
see that there might be statistically significant difference between
different cohorts.</p>
<p>Let&#x2019;s compare Shannon index of ADHD samples between cohort 2 and cohort
3.</p>
<p>As we can see, there is statistically significant difference between the
cohorts.</p>
<pre><code># Takes subset of colData. Takes only ADHD samples
ADHD_shannon &lt;- colData(tse)[ colData(tse)[, &quot;patient_status&quot;] == &quot;ADHD&quot; , ]

# Takes subset of colData. Takes only samples that are in cohort 2 or cohort 3.
ADHD_shannon &lt;- ADHD_shannon[ ADHD_shannon[, &quot;cohort&quot;] %in% c(&quot;Cohort_2&quot;, &quot;Cohort_3&quot;) , ]

# Wilcoxon test, where Shannon index is the variable that we are comparing. 
# Cohort - 2 or 3 - is the factor that we use for grouping. 
wilcoxon_shannon_ADHD_cohorts &lt;- wilcox.test(Shannon_index ~ cohort, data = ADHD_shannon)

wilcoxon_shannon_ADHD_cohorts

## 
##  Wilcoxon rank sum exact test
## 
## data:  Shannon_index by cohort
## W = 20, p-value = 0.01587
## alternative hypothesis: true location shift is not equal to 0
</code></pre><h2 id="further-resources">Further resources</h2>
<p>For more examples, see a dedicated section on alpha diversity in the
<a href="https://microbiome.github.io/OMA/" target="_blank">online book</a>.</p>

                                
                                </section>
                            
    </div>
    <div class="search-results">
        <div class="has-results">
            
            <h1 class="search-results-title"><span class='search-results-count'></span> results matching "<span class='search-query'></span>"</h1>
            <ul class="search-results-list"></ul>
            
        </div>
        <div class="no-results">
            
            <h1 class="search-results-title">No results matching "<span class='search-query'></span>"</h1>
            
        </div>
    </div>
</div>

                        </div>
                    </div>
                
            </div>

            
                
                <a href="explore.html" class="navigation navigation-prev " aria-label="Previous page: Explore data">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
                <a href="beta.html" class="navigation navigation-next " aria-label="Next page: Calculate beta diversity">
                    <i class="fa fa-angle-right"></i>
                </a>
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"Calculate alpha diversity","level":"1.5","depth":1,"next":{"title":"Calculate beta diversity","level":"1.6","depth":1,"path":"beta.md","ref":"beta.md","articles":[]},"previous":{"title":"Explore data","level":"1.4","depth":1,"path":"explore.md","ref":"explore.md","articles":[]},"dir":"ltr"},"config":{"gitbook":"*","theme":"default","variables":{},"plugins":[],"pluginsConfig":{"highlight":{},"search":{},"lunr":{"maxIndexSize":1000000,"ignoreSpecialCharacters":false},"sharing":{"facebook":true,"twitter":true,"google":false,"weibo":false,"instapaper":false,"vk":false,"all":["facebook","google","twitter","weibo","instapaper"]},"fontsettings":{"theme":"white","family":"sans","size":2},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":false}},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"pdf":{"pageNumbers":true,"fontSize":12,"fontFamily":"Arial","paperSize":"a4","chapterMark":"pagebreak","pageBreaksBefore":"/","margin":{"right":62,"left":62,"top":56,"bottom":56}},"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"}},"file":{"path":"alpha.md","mtime":"2021-06-09T07:09:09.355Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2021-06-09T08:40:16.093Z"},"basePath":".","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="gitbook/gitbook.js"></script>
    <script src="gitbook/theme.js"></script>
    
        
        <script src="gitbook/gitbook-plugin-search/search-engine.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-search/search.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-lunr/lunr.min.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-lunr/search-lunr.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-sharing/buttons.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

