<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Beta diversity | Introduction to microbiome data science</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Beta diversity | Introduction to microbiome data science" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/course_2021_radboud" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Beta diversity | Introduction to microbiome data science" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="Leo Lahti, Tuomas Borman, and Henrik Eckermann" />


<meta name="date" content="2021-07-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="alpha-diversity.html"/>
<link rel="next" href="community-typing.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to miaverse</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-miaverse-framework"><i class="fa fa-check"></i><b>1.1</b> The miaverse framework</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#learning-goals"><i class="fa fa-check"></i><b>1.2</b> Learning goals</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="program.html"><a href="program.html"><i class="fa fa-check"></i><b>2</b> Program</a>
<ul>
<li class="chapter" data-level="2.1" data-path="program.html"><a href="program.html#monday-12-july-from-raw-sequences-to-ecological-data-analysis"><i class="fa fa-check"></i><b>2.1</b> Monday 12 July: from raw sequences to ecological data analysis</a></li>
<li class="chapter" data-level="2.2" data-path="program.html"><a href="program.html#tuesday-13-july---alpha-diversity"><i class="fa fa-check"></i><b>2.2</b> Tuesday 13 July - Alpha diversity</a></li>
<li class="chapter" data-level="2.3" data-path="program.html"><a href="program.html#wednesday-14-july---beta-diversity"><i class="fa fa-check"></i><b>2.3</b> Wednesday 14 July - Beta diversity</a></li>
<li class="chapter" data-level="2.4" data-path="program.html"><a href="program.html#thursday-15-july---differential-abundance"><i class="fa fa-check"></i><b>2.4</b> Thursday 15 July - Differential abundance</a></li>
<li class="chapter" data-level="2.5" data-path="program.html"><a href="program.html#friday-16-july-presentations-closing"><i class="fa fa-check"></i><b>2.5</b> Friday 16 July: Presentations &amp; closing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#checklist-before-the-course"><i class="fa fa-check"></i><b>3.1</b> Checklist (before the course)</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#support-and-resources"><i class="fa fa-check"></i><b>3.2</b> Support and resources</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#installing-and-loading-the-required-r-packages"><i class="fa fa-check"></i><b>3.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html"><i class="fa fa-check"></i><b>4</b> Importing microbiome data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#data-access-and-example-of-importing-data"><i class="fa fa-check"></i><b>4.1</b> Data access and example of importing data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html"><i class="fa fa-check"></i><b>5</b> Microbiome data exploration</a>
<ul>
<li class="chapter" data-level="5.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#investigate"><i class="fa fa-check"></i><b>5.1</b> Investigate</a></li>
<li class="chapter" data-level="5.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#manipulate"><i class="fa fa-check"></i><b>5.2</b> Manipulate</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#transformations"><i class="fa fa-check"></i><b>5.2.1</b> Transformations</a></li>
<li class="chapter" data-level="5.2.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#aggregation"><i class="fa fa-check"></i><b>5.2.2</b> Aggregation</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#visualize"><i class="fa fa-check"></i><b>5.3</b> Visualize</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>6</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#visualization"><i class="fa fa-check"></i><b>6.1</b> Visualization</a></li>
<li class="chapter" data-level="6.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#statistical-testing-and-comparisons"><i class="fa fa-check"></i><b>6.2</b> Statistical testing and comparisons</a></li>
<li class="chapter" data-level="6.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#further-resources"><i class="fa fa-check"></i><b>6.3</b> Further resources</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>7</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#examples-of-pcoa-with-different-settings"><i class="fa fa-check"></i><b>7.1</b> Examples of PCoA with different settings</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-bray-curtis"><i class="fa fa-check"></i><b>7.1.1</b> PCoA for ASV-level data with Bray-Curtis</a></li>
<li class="chapter" data-level="7.1.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-aitchison-distance"><i class="fa fa-check"></i><b>7.1.2</b> PCoA for ASV-level data with Aitchison distance</a></li>
<li class="chapter" data-level="7.1.3" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-aggregated-to-phylum-level"><i class="fa fa-check"></i><b>7.1.3</b> PCoA aggregated to Phylum level</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#highlighting-external-variables-on-pcoa-plot"><i class="fa fa-check"></i><b>7.2</b> Highlighting external variables on PCoA plot</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-with-discrete-sample-grouping-variable-shown-with-colors"><i class="fa fa-check"></i><b>7.2.1</b> PCoA with discrete sample grouping variable shown with colors</a></li>
<li class="chapter" data-level="7.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-plot-with-continuous-variable"><i class="fa fa-check"></i><b>7.2.2</b> PCoA plot with continuous variable</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#estimating-associations-with-an-external-variable"><i class="fa fa-check"></i><b>7.3</b> Estimating associations with an external variable</a></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#further-resources-1"><i class="fa fa-check"></i><b>7.4</b> Further resources</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-typing.html"><a href="community-typing.html"><i class="fa fa-check"></i><b>8</b> Community typing</a>
<ul>
<li class="chapter" data-level="8.1" data-path="community-typing.html"><a href="community-typing.html#dirichlet-multinomial-mixture-model"><i class="fa fa-check"></i><b>8.1</b> Dirichlet-Multinomial Mixture Model</a></li>
<li class="chapter" data-level="8.2" data-path="community-typing.html"><a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>8.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html"><i class="fa fa-check"></i><b>9</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#wilcoxon-test"><i class="fa fa-check"></i><b>9.1</b> Wilcoxon test</a></li>
<li class="chapter" data-level="9.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#deseq2"><i class="fa fa-check"></i><b>9.2</b> DESeq2</a></li>
<li class="chapter" data-level="9.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#ancom-bc"><i class="fa fa-check"></i><b>9.3</b> ANCOM-BC</a></li>
<li class="chapter" data-level="9.4" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-wilcoxon-test-and-deseq2"><i class="fa fa-check"></i><b>9.4</b> Comparison of Wilcoxon test and DESeq2</a></li>
<li class="chapter" data-level="9.5" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-abundance"><i class="fa fa-check"></i><b>9.5</b> Comparison of abundance</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="study-material.html"><a href="study-material.html"><i class="fa fa-check"></i><b>10</b> Study material</a>
<ul>
<li class="chapter" data-level="10.1" data-path="study-material.html"><a href="study-material.html#lecture-slides"><i class="fa fa-check"></i><b>10.1</b> Lecture slides</a></li>
<li class="chapter" data-level="10.2" data-path="study-material.html"><a href="study-material.html#example-solutions"><i class="fa fa-check"></i><b>10.2</b> Example solutions</a></li>
<li class="chapter" data-level="10.3" data-path="study-material.html"><a href="study-material.html#r-programming-resources"><i class="fa fa-check"></i><b>10.3</b> R programming resources</a></li>
<li class="chapter" data-level="10.4" data-path="study-material.html"><a href="study-material.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>10.4</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="10.5" data-path="study-material.html"><a href="study-material.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>10.5</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="10.6" data-path="study-material.html"><a href="study-material.html#further-reading"><i class="fa fa-check"></i><b>10.6</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html"><i class="fa fa-check"></i><b>11</b> Miscellaneous material</a>
<ul>
<li class="chapter" data-level="11.1" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html#shapiro-wilk-test"><i class="fa fa-check"></i><b>11.1</b> Shapiro-Wilk test</a></li>
<li class="chapter" data-level="11.2" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html#deseq-details"><i class="fa fa-check"></i><b>11.2</b> Deseq details</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to microbiome data science</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="beta-diversity" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Beta diversity</h1>
<p>In this section, we go through how to calculate and visualize beta diversity.</p>
<p>Beta diversity reflects the difference in microbial composition
between samples. Unlike alpha diversity, where we calculate diversity within a sample, any particular beta diversity value can only be calculated between two samples. There are different ways to measure the distance between two samples. Some of the common choices include Bray-Curtis, Unifrac, Jaccard index, and the Aitchison distance metric. Each of these (dis)similarity measures emphasizes different aspects of (dis)similarity. For example, UniFrac is a distance metric that incorporates phylogenetic information between samples. Or, the Jaccard-index measures the distance based on presence/absence of any taxon irrespective of its abundance.</p>
<div id="examples-of-pcoa-with-different-settings" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Examples of PCoA with different settings</h2>
<p>After calculating beta diversity, we end up with a (dis)similarity matrix that contains for each sample the distance to any other sample. To explore our data, we would like to visualize the distances between samples and potentially map meta data into the same graph. This would enable us to explore whether distance between samples could be related to known variables. Given the many dimensions in the matrix, we first need to apply a technique that summarises most of the information into fewer dimensions (components). Principal Coordinate Analysis (PCoA) is a technique that can achieve exactly that.</p>
<p>As a side note for those of you familiar with Principal Component Analysis (PCA): A PCoA where you enter a matrix of Euclidean distances will be similar to a PCA of the microbial abundances (except for potential scaling differences).</p>
<p>Back to our PCoA: Typically, we retain 2-3 components that contain most of the information in the data. Note however, that the other components might contain important information as well. After all these steps, we thus end up with 2-3 new variables (components), where each sample has a score on each of those components. We can then simply create a 2-D or a 3D plot with the components on the x and y (and z) axis. We could do the same for any combination of the many components a PCoA calculates. Enough theory, let’s try to put this into practice!</p>
<div id="pcoa-for-asv-level-data-with-bray-curtis" class="section level3" number="7.1.1">
<h3><span class="header-section-number">7.1.1</span> PCoA for ASV-level data with Bray-Curtis</h3>
<p>We will illustrate what we described above using different distance measures and also applied to different taxonomic ranks. We start with a PCoA that uses a Bray Curtis matrix as input, calculated at the Genus level.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="beta-diversity.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Relative abundance table</span></span>
<span id="cb36-2"><a href="beta-diversity.html#cb36-2" aria-hidden="true" tabindex="-1"></a>rel_abund_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse)<span class="sc">$</span>relabundance</span>
<span id="cb36-3"><a href="beta-diversity.html#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposes it to get taxa to columns</span></span>
<span id="cb36-4"><a href="beta-diversity.html#cb36-4" aria-hidden="true" tabindex="-1"></a>rel_abund_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(rel_abund_assay)</span>
<span id="cb36-5"><a href="beta-diversity.html#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates Bray-Curtis distances between samples. Because taxa is in columns,</span></span>
<span id="cb36-6"><a href="beta-diversity.html#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="co"># it is used to compare different samples.</span></span>
<span id="cb36-7"><a href="beta-diversity.html#cb36-7" aria-hidden="true" tabindex="-1"></a>bray_curtis_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(rel_abund_assay, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb36-8"><a href="beta-diversity.html#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Does principal coordinate analysis</span></span>
<span id="cb36-9"><a href="beta-diversity.html#cb36-9" aria-hidden="true" tabindex="-1"></a>bray_curtis_pcoa <span class="ot">&lt;-</span> ecodist<span class="sc">::</span><span class="fu">pco</span>(bray_curtis_dist)</span>
<span id="cb36-10"><a href="beta-diversity.html#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="co"># all components could be found here: </span></span>
<span id="cb36-11"><a href="beta-diversity.html#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="co"># bray_curtis_pcoa$vectors</span></span>
<span id="cb36-12"><a href="beta-diversity.html#cb36-12" aria-hidden="true" tabindex="-1"></a><span class="co"># But we only need the first two to demonstrate what we can do:</span></span>
<span id="cb36-13"><a href="beta-diversity.html#cb36-13" aria-hidden="true" tabindex="-1"></a>bray_curtis_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pcoa1 =</span> bray_curtis_pcoa<span class="sc">$</span>vectors[,<span class="dv">1</span>], </span>
<span id="cb36-14"><a href="beta-diversity.html#cb36-14" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">pcoa2 =</span> bray_curtis_pcoa<span class="sc">$</span>vectors[,<span class="dv">2</span>])</span>
<span id="cb36-15"><a href="beta-diversity.html#cb36-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-16"><a href="beta-diversity.html#cb36-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb36-17"><a href="beta-diversity.html#cb36-17" aria-hidden="true" tabindex="-1"></a>bray_curtis_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> bray_curtis_pcoa_df, <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2)) <span class="sc">+</span></span>
<span id="cb36-18"><a href="beta-diversity.html#cb36-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb36-19"><a href="beta-diversity.html#cb36-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Coordinate 1&quot;</span>,</span>
<span id="cb36-20"><a href="beta-diversity.html#cb36-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coordinate 2&quot;</span>, </span>
<span id="cb36-21"><a href="beta-diversity.html#cb36-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;Bray-Curtis PCoA with relative abundances&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-22"><a href="beta-diversity.html#cb36-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb36-23"><a href="beta-diversity.html#cb36-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-24"><a href="beta-diversity.html#cb36-24" aria-hidden="true" tabindex="-1"></a>bray_curtis_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/pcoa_asv_bc-1.png" width="672" /></p>
</div>
<div id="pcoa-for-asv-level-data-with-aitchison-distance" class="section level3" number="7.1.2">
<h3><span class="header-section-number">7.1.2</span> PCoA for ASV-level data with Aitchison distance</h3>
<p>Now the same using Aitchison distance. This metric corresponds to Euclidean distances between CLR transformed sample abundance vectors.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="beta-diversity.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr transformation. Pseudocount is added, because data contains zeros. </span></span>
<span id="cb37-2"><a href="beta-diversity.html#cb37-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb37-3"><a href="beta-diversity.html#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets clr table</span></span>
<span id="cb37-4"><a href="beta-diversity.html#cb37-4" aria-hidden="true" tabindex="-1"></a>clr_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse)<span class="sc">$</span>clr</span>
<span id="cb37-5"><a href="beta-diversity.html#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposes it to get taxa to columns</span></span>
<span id="cb37-6"><a href="beta-diversity.html#cb37-6" aria-hidden="true" tabindex="-1"></a>clr_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(clr_assay)</span>
<span id="cb37-7"><a href="beta-diversity.html#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates Euclidean distances between samples. Because taxa is in columns,</span></span>
<span id="cb37-8"><a href="beta-diversity.html#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="co"># it is used to compare different samples.</span></span>
<span id="cb37-9"><a href="beta-diversity.html#cb37-9" aria-hidden="true" tabindex="-1"></a>euclidean_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(clr_assay, <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb37-10"><a href="beta-diversity.html#cb37-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Does principal coordinate analysis</span></span>
<span id="cb37-11"><a href="beta-diversity.html#cb37-11" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa <span class="ot">&lt;-</span> ecodist<span class="sc">::</span><span class="fu">pco</span>(euclidean_dist)</span>
<span id="cb37-12"><a href="beta-diversity.html#cb37-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame from principal coordinates</span></span>
<span id="cb37-13"><a href="beta-diversity.html#cb37-13" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pcoa1 =</span> euclidean_pcoa<span class="sc">$</span>vectors[,<span class="dv">1</span>], </span>
<span id="cb37-14"><a href="beta-diversity.html#cb37-14" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pcoa2 =</span> euclidean_pcoa<span class="sc">$</span>vectors[,<span class="dv">2</span>])</span>
<span id="cb37-15"><a href="beta-diversity.html#cb37-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-16"><a href="beta-diversity.html#cb37-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb37-17"><a href="beta-diversity.html#cb37-17" aria-hidden="true" tabindex="-1"></a>euclidean_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_pcoa_df, <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2)) <span class="sc">+</span></span>
<span id="cb37-18"><a href="beta-diversity.html#cb37-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb37-19"><a href="beta-diversity.html#cb37-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Coordinate 1&quot;</span>,</span>
<span id="cb37-20"><a href="beta-diversity.html#cb37-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coordinate 2&quot;</span>,</span>
<span id="cb37-21"><a href="beta-diversity.html#cb37-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;Euclidean PCoA with CLR transformation&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-22"><a href="beta-diversity.html#cb37-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb37-23"><a href="beta-diversity.html#cb37-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-24"><a href="beta-diversity.html#cb37-24" aria-hidden="true" tabindex="-1"></a>euclidean_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/pcoa_asv_aitchison-1.png" width="672" /></p>
</div>
<div id="pcoa-aggregated-to-phylum-level" class="section level3" number="7.1.3">
<h3><span class="header-section-number">7.1.3</span> PCoA aggregated to Phylum level</h3>
<p>We use again the Aitchison distances in this example but this time applied to the phylum level.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="beta-diversity.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr transformation. Psuedocount is added, because data contains zeros. </span></span>
<span id="cb38-2"><a href="beta-diversity.html#cb38-2" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse_phylum, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb38-3"><a href="beta-diversity.html#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets clr table</span></span>
<span id="cb38-4"><a href="beta-diversity.html#cb38-4" aria-hidden="true" tabindex="-1"></a>clr_phylum_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse_phylum)<span class="sc">$</span>clr</span>
<span id="cb38-5"><a href="beta-diversity.html#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposes it to get taxa to columns</span></span>
<span id="cb38-6"><a href="beta-diversity.html#cb38-6" aria-hidden="true" tabindex="-1"></a>clr_phylum_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(clr_phylum_assay)</span>
<span id="cb38-7"><a href="beta-diversity.html#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates Euclidean distances between samples. Because taxa is in columns,</span></span>
<span id="cb38-8"><a href="beta-diversity.html#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="co"># it is used to compare different samples.</span></span>
<span id="cb38-9"><a href="beta-diversity.html#cb38-9" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(clr_assay, <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb38-10"><a href="beta-diversity.html#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Does principal coordinate analysis</span></span>
<span id="cb38-11"><a href="beta-diversity.html#cb38-11" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_pcoa <span class="ot">&lt;-</span> ecodist<span class="sc">::</span><span class="fu">pco</span>(euclidean_phylum_dist)</span>
<span id="cb38-12"><a href="beta-diversity.html#cb38-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame from principal coordinates</span></span>
<span id="cb38-13"><a href="beta-diversity.html#cb38-13" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pcoa1 =</span> euclidean_phylum_pcoa<span class="sc">$</span>vectors[,<span class="dv">1</span>], </span>
<span id="cb38-14"><a href="beta-diversity.html#cb38-14" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">pcoa2 =</span> euclidean_phylum_pcoa<span class="sc">$</span>vectors[,<span class="dv">2</span>])</span>
<span id="cb38-15"><a href="beta-diversity.html#cb38-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-16"><a href="beta-diversity.html#cb38-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb38-17"><a href="beta-diversity.html#cb38-17" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_phylum_pcoa_df, <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2)) <span class="sc">+</span></span>
<span id="cb38-18"><a href="beta-diversity.html#cb38-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb38-19"><a href="beta-diversity.html#cb38-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Coordinate 1&quot;</span>,</span>
<span id="cb38-20"><a href="beta-diversity.html#cb38-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coordinate 2&quot;</span>,</span>
<span id="cb38-21"><a href="beta-diversity.html#cb38-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;Aitchison distances at Phylum level&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb38-22"><a href="beta-diversity.html#cb38-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb38-23"><a href="beta-diversity.html#cb38-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-24"><a href="beta-diversity.html#cb38-24" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/pcoa_phylum_aitchison-1.png" width="672" /></p>
</div>
</div>
<div id="highlighting-external-variables-on-pcoa-plot" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Highlighting external variables on PCoA plot</h2>
<p>As explained in the introduction, it can help to explore the data if we map other variables on the same plot. For example, by changing the color of the points based on that variable. Let’s see how we could do that:</p>
<div id="pcoa-with-discrete-sample-grouping-variable-shown-with-colors" class="section level3" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> PCoA with discrete sample grouping variable shown with colors</h3>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="beta-diversity.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds the variable we later use for coloring to the data frame</span></span>
<span id="cb39-2"><a href="beta-diversity.html#cb39-2" aria-hidden="true" tabindex="-1"></a>euclidean_patient_status_pcoa_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(euclidean_pcoa_df,</span>
<span id="cb39-3"><a href="beta-diversity.html#cb39-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">patient_status =</span> <span class="fu">colData</span>(tse)<span class="sc">$</span>patient_status)</span>
<span id="cb39-4"><a href="beta-diversity.html#cb39-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-5"><a href="beta-diversity.html#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb39-6"><a href="beta-diversity.html#cb39-6" aria-hidden="true" tabindex="-1"></a>euclidean_patient_status_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_patient_status_pcoa_df, </span>
<span id="cb39-7"><a href="beta-diversity.html#cb39-7" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2,</span>
<span id="cb39-8"><a href="beta-diversity.html#cb39-8" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">color =</span> patient_status)) <span class="sc">+</span></span>
<span id="cb39-9"><a href="beta-diversity.html#cb39-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb39-10"><a href="beta-diversity.html#cb39-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Coordinate 1&quot;</span>,</span>
<span id="cb39-11"><a href="beta-diversity.html#cb39-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coordinate 2&quot;</span>,</span>
<span id="cb39-12"><a href="beta-diversity.html#cb39-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;PCoA with Aitchison distances&quot;</span>) <span class="sc">+</span></span>
<span id="cb39-13"><a href="beta-diversity.html#cb39-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb39-14"><a href="beta-diversity.html#cb39-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-15"><a href="beta-diversity.html#cb39-15" aria-hidden="true" tabindex="-1"></a>euclidean_patient_status_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/pcoa_genus-1.png" width="672" /></p>
</div>
<div id="pcoa-plot-with-continuous-variable" class="section level3" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> PCoA plot with continuous variable</h3>
<p>We can do the same as above using any continuous variable. E.g. let us see how the plotted samples differ in their alpha diversity:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="beta-diversity.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds coloring information to the data frame, creates new column</span></span>
<span id="cb40-2"><a href="beta-diversity.html#cb40-2" aria-hidden="true" tabindex="-1"></a>euclidean_shannon_pcoa_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(euclidean_pcoa_df,</span>
<span id="cb40-3"><a href="beta-diversity.html#cb40-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">shannon =</span> <span class="fu">colData</span>(tse)<span class="sc">$</span>Shannon_index)</span>
<span id="cb40-4"><a href="beta-diversity.html#cb40-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-5"><a href="beta-diversity.html#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb40-6"><a href="beta-diversity.html#cb40-6" aria-hidden="true" tabindex="-1"></a>euclidean_shannon_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_shannon_pcoa_df, </span>
<span id="cb40-7"><a href="beta-diversity.html#cb40-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2,</span>
<span id="cb40-8"><a href="beta-diversity.html#cb40-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">color =</span> shannon)) <span class="sc">+</span> </span>
<span id="cb40-9"><a href="beta-diversity.html#cb40-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb40-10"><a href="beta-diversity.html#cb40-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Coordinate 1&quot;</span>,</span>
<span id="cb40-11"><a href="beta-diversity.html#cb40-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;Coordinate 2&quot;</span>,</span>
<span id="cb40-12"><a href="beta-diversity.html#cb40-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;PCoA with Aitchison distances&quot;</span>) <span class="sc">+</span></span>
<span id="cb40-13"><a href="beta-diversity.html#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb40-14"><a href="beta-diversity.html#cb40-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-15"><a href="beta-diversity.html#cb40-15" aria-hidden="true" tabindex="-1"></a>euclidean_shannon_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/pcoa_coloring-1.png" width="672" /></p>
</div>
</div>
<div id="estimating-associations-with-an-external-variable" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Estimating associations with an external variable</h2>
<p>Next to visualizing whether any variable is associated with differences between samples, we can also quantify the strength of the association between the
variation in community composition (beta diversity) and external
factors. The current standard way to do this is to perform a so-called permutational multivariate analysis of variance (PERMANOVA). This method takes as input the ASV/OTU table, which measure of distance you want to base the test on and a formula that tells the model how you think the variables are associated with each other. Let’s try it:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="beta-diversity.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First we get the relative abundance table</span></span>
<span id="cb41-2"><a href="beta-diversity.html#cb41-2" aria-hidden="true" tabindex="-1"></a>rel_abund_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse)<span class="sc">$</span>relabundance</span>
<span id="cb41-3"><a href="beta-diversity.html#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="co"># again transpose it to get taxa to columns</span></span>
<span id="cb41-4"><a href="beta-diversity.html#cb41-4" aria-hidden="true" tabindex="-1"></a>rel_abund_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(rel_abund_assay)</span>
<span id="cb41-5"><a href="beta-diversity.html#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="co"># then we can perform the method</span></span>
<span id="cb41-6"><a href="beta-diversity.html#cb41-6" aria-hidden="true" tabindex="-1"></a>permanova_cohort <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis</span>(rel_abund_assay <span class="sc">~</span> cohort,</span>
<span id="cb41-7"><a href="beta-diversity.html#cb41-7" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> <span class="fu">colData</span>(tse),</span>
<span id="cb41-8"><a href="beta-diversity.html#cb41-8" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">permutations =</span> <span class="dv">9999</span>)</span>
<span id="cb41-9"><a href="beta-diversity.html#cb41-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-10"><a href="beta-diversity.html#cb41-10" aria-hidden="true" tabindex="-1"></a><span class="co"># we can obtain a the p value for our predictor:</span></span>
<span id="cb41-11"><a href="beta-diversity.html#cb41-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;Different different cohorts and variance of abundance between samples, p-value: &quot;</span>, </span>
<span id="cb41-12"><a href="beta-diversity.html#cb41-12" aria-hidden="true" tabindex="-1"></a>             <span class="fu">as.data.frame</span>(permanova_cohort<span class="sc">$</span>aov.tab)[<span class="st">&quot;cohort&quot;</span>, <span class="st">&quot;Pr(&gt;F)&quot;</span>]))</span></code></pre></div>
<pre><code>## [1] &quot;Different different cohorts and variance of abundance between samples, p-value: 0.7378&quot;</code></pre>
<p>As we see, the cohort variable is not significantly associated with
microbiota composition (p-value is over 0.05).</p>
<p>We can still visualize those taxa whose abundances are the most
different between cohorts. This gives us information how taxonomic abundances tend to differ between different cohorts. In order to do that, we first need to extract the model coefficients of taxa:</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="beta-diversity.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets the coefficients</span></span>
<span id="cb43-2"><a href="beta-diversity.html#cb43-2" aria-hidden="true" tabindex="-1"></a>coef <span class="ot">&lt;-</span> <span class="fu">coefficients</span>(permanova_cohort)[<span class="st">&quot;cohort1&quot;</span>,]</span>
<span id="cb43-3"><a href="beta-diversity.html#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets the highest coefficients</span></span>
<span id="cb43-4"><a href="beta-diversity.html#cb43-4" aria-hidden="true" tabindex="-1"></a>top.coef <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">head</span>(coef[<span class="fu">rev</span>(<span class="fu">order</span>(<span class="fu">abs</span>(coef)))],<span class="dv">20</span>))</span>
<span id="cb43-5"><a href="beta-diversity.html#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plots the coefficients</span></span>
<span id="cb43-6"><a href="beta-diversity.html#cb43-6" aria-hidden="true" tabindex="-1"></a>top_taxa_coeffient_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> top.coef,</span>
<span id="cb43-7"><a href="beta-diversity.html#cb43-7" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">y =</span> <span class="fu">factor</span>(<span class="fu">names</span>(top.coef),</span>
<span id="cb43-8"><a href="beta-diversity.html#cb43-8" aria-hidden="true" tabindex="-1"></a>                                                        <span class="fu">unique</span>(<span class="fu">names</span>(top.coef)))),</span>
<span id="cb43-9"><a href="beta-diversity.html#cb43-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y)) <span class="sc">+</span></span>
<span id="cb43-10"><a href="beta-diversity.html#cb43-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb43-11"><a href="beta-diversity.html#cb43-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;&quot;</span>, <span class="at">y=</span><span class="st">&quot;&quot;</span>, <span class="at">title=</span><span class="st">&quot;Top Taxa&quot;</span>) <span class="sc">+</span></span>
<span id="cb43-12"><a href="beta-diversity.html#cb43-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span>
<span id="cb43-13"><a href="beta-diversity.html#cb43-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-14"><a href="beta-diversity.html#cb43-14" aria-hidden="true" tabindex="-1"></a>top_taxa_coeffient_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<p>The above plot shows taxa as code names, and it is hard to tell which
bacterial groups they represent. However, it is easy to add human readable
names. We can fetch those from our rowData. Here we use Genus level names:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="beta-diversity.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets corresponding Genus level names and stores them to top.coef</span></span>
<span id="cb44-2"><a href="beta-diversity.html#cb44-2" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">rowData</span>(tse)[<span class="fu">names</span>(top.coef), ][,<span class="st">&quot;Genus&quot;</span>]</span>
<span id="cb44-3"><a href="beta-diversity.html#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds new labels to the plot</span></span>
<span id="cb44-4"><a href="beta-diversity.html#cb44-4" aria-hidden="true" tabindex="-1"></a>top_taxa_coeffient_plot <span class="ot">&lt;-</span> top_taxa_coeffient_plot <span class="sc">+</span></span>
<span id="cb44-5"><a href="beta-diversity.html#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> names) <span class="co"># Adds new labels</span></span>
<span id="cb44-6"><a href="beta-diversity.html#cb44-6" aria-hidden="true" tabindex="-1"></a>top_taxa_coeffient_plot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
</div>
<div id="further-resources-1" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Further resources</h2>
<p>For more examples, see a dedicated section on beta diversity in the
<a href="https://microbiome.github.io/OMA/">online book</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="alpha-diversity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="community-typing.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
