
<!DOCTYPE HTML>
<html lang="" >
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>DMM analysis Â· GitBook</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="description" content="">
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-highlight/website.css">
                
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-search/search.css">
                
            
                
                <link rel="stylesheet" href="gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    
        
    
        
    
        
    
        
    
        
    
        
    

        
    
    
    <meta name="HandheldFriendly" content="true"/>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="gitbook/images/favicon.ico" type="image/x-icon">

    
    <link rel="next" href="dmm.html" />
    
    
    <link rel="prev" href="beta.html" />
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary">
        
            
<div id="book-search-input" role="search">
    <input type="text" placeholder="Type to search" />
</div>

            
                <nav role="navigation">
                


<ul class="summary">
    
    

    

    
        
        
    
        <li class="chapter " data-level="1.1" data-path="./">
            
                <a href="./">
            
                    
                    Introduction
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.1.1" data-path="./">
            
                <a href="./#miaverse">
            
                    
                    miaverse
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.1.2" data-path="./">
            
                <a href="./#acknowledgments">
            
                    
                    Acknowledgments
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.2" data-path="start.html">
            
                <a href="start.html">
            
                    
                    Getting started
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.2.1" data-path="start.html">
            
                <a href="start.html#checklist-before-the-course">
            
                    
                    Checklist before the course
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2.2" data-path="start.html">
            
                <a href="start.html#start-workflow">
            
                    
                    Start workflow
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2.3" data-path="start.html">
            
                <a href="start.html#online-support">
            
                    
                    Online support
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2.4" data-path="start.html">
            
                <a href="start.html#additional-resources">
            
                    
                    Additional resources
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.3" data-path="install.html">
            
                <a href="install.html">
            
                    
                    Install and load required R packages
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4" data-path="import.html">
            
                <a href="import.html">
            
                    
                    Import data
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.1" data-path="import.html">
            
                <a href="import.html#data">
            
                    
                    Data
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2" data-path="import.html">
            
                <a href="import.html#investigate-the-r-data-object">
            
                    
                    Investigate the R data object
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3" data-path="import.html">
            
                <a href="import.html#rowdata-taxonomic-information">
            
                    
                    rowData (taxonomic information)
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.4" data-path="import.html">
            
                <a href="import.html#coldata-sample-information">
            
                    
                    colData (sample information)
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.5" data-path="import.html">
            
                <a href="import.html#phylogenetic-tree-information">
            
                    
                    Phylogenetic tree information
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.5" data-path="explore.html">
            
                <a href="explore.html">
            
                    
                    Explore data
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.5.1" data-path="explore.html">
            
                <a href="explore.html#investigate-the-contents-of-a-microbiome-data-object">
            
                    
                    Investigate the contents of a microbiome data object
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2" data-path="explore.html">
            
                <a href="explore.html#manipulate-the-data">
            
                    
                    Manipulate the data
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.5.2.1" data-path="explore.html">
            
                <a href="explore.html#transformations">
            
                    
                    Transformations
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2.2" data-path="explore.html">
            
                <a href="explore.html#aggregation">
            
                    
                    Aggregation
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.5.3" data-path="explore.html">
            
                <a href="explore.html#visualization">
            
                    
                    Visualization
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.4" data-path="explore.html">
            
                <a href="explore.html#further-resources">
            
                    
                    Further resources
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.6" data-path="alpha.html">
            
                <a href="alpha.html">
            
                    
                    Calculate alpha diversity
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.6.1" data-path="alpha.html">
            
                <a href="alpha.html#visualization">
            
                    
                    Visualization
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.6.2" data-path="alpha.html">
            
                <a href="alpha.html#statistical-testing-and-comparisons">
            
                    
                    Statistical testing and comparisons
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.6.3" data-path="alpha.html">
            
                <a href="alpha.html#further-resources">
            
                    
                    Further resources
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.7" data-path="beta.html">
            
                <a href="beta.html">
            
                    
                    Calculate beta diversity
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.7.1" data-path="beta.html">
            
                <a href="beta.html#examples-of-pcoa-with-different-settings">
            
                    
                    Examples of PCoA with different settings
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.7.1.1" data-path="beta.html">
            
                <a href="beta.html#pcoa-for-asv-level-data-with-bray-curtis">
            
                    
                    PCoA for ASV-level data with Bray-Curtis
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.1.2" data-path="beta.html">
            
                <a href="beta.html#pcoa-for-asv-level-data-with-aitchison-distance">
            
                    
                    PCoA for ASV-level data with Aitchison distance
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.1.3" data-path="beta.html">
            
                <a href="beta.html#pcoa-aggregated-to-phylum-level">
            
                    
                    PCoA aggregated to Phylum level
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.7.2" data-path="beta.html">
            
                <a href="beta.html#highlighting-external-variables-on-pcoa-plot">
            
                    
                    Highlighting external variables on PCoA plot
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.7.2.1" data-path="beta.html">
            
                <a href="beta.html#pcoa-with-discrete-sample-grouping-variable-shown-with-colors">
            
                    
                    PCoA with discrete sample grouping variable shown with colors
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.2.2" data-path="beta.html">
            
                <a href="beta.html#pcoa-with-continuous-sample-grouping-variable-shown-with-colors">
            
                    
                    PCoA with continuous sample grouping variable shown with colors
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.7.3" data-path="beta.html">
            
                <a href="beta.html#estimating-associations-with-an-external-variable">
            
                    
                    Estimating associations with an external variable
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.4" data-path="beta.html">
            
                <a href="beta.html#further-resources">
            
                    
                    Further resources
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter active" data-level="1.8" data-path="dmm.html">
            
                <a href="dmm.html">
            
                    
                    DMM analysis
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.8.1" data-path="dmm.html">
            
                <a href="dmm.html#pcoa-for-asv-level-data-with-bray-curtis;-with-dmm-clusters-shown-with-colors">
            
                    
                    PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.9" data-path="abundance.html">
            
                <a href="abundance.html">
            
                    
                    Differential abundance analysis
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.9.1" data-path="abundance.html">
            
                <a href="abundance.html#wilcoxon-test">
            
                    
                    Wilcoxon test
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.9.2" data-path="abundance.html">
            
                <a href="abundance.html#deseq2">
            
                    
                    DESeq2
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.9.3" data-path="abundance.html">
            
                <a href="abundance.html#ancom-bc">
            
                    
                    ANCOM-BC
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.9.4" data-path="abundance.html">
            
                <a href="abundance.html#comparison-of-wilcoxon-test-and-deseq2">
            
                    
                    Comparison of Wilcoxon test and DESeq2
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.9.5" data-path="abundance.html">
            
                <a href="abundance.html#comparison-of-abundance">
            
                    
                    Comparison of abundance
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            Published with GitBook
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href="." >DMM analysis</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
<div id="book-search-results">
    <div class="search-noresults">
    
                                <section class="normal markdown-section">
                                
                                <h1 id="dirichlet-multinomial-mixture-model">Dirichlet-Multinomial Mixture Model</h1>
<p>This section focus on DMM analysis.</p>
<p>One technique that allows to search for groups of samples that are
similar to each other is the <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0030126" target="_blank">Dirichlet-Multinomial Mixture
Model</a>.
In DMM, we first determine the number of clusters (k) that best fit the
data (model evidence) using Laplace approximation. After fitting the
model with k clusters, we obtain for each sample k probabilities that
reflect the probability that a sample belongs to the given cluster.</p>
<p>Let&#x2019;s cluster the data with DMM clustering.</p>
<pre><code># Runs model and calculates the most likely number of clusters from 1 to 7. 
# For this small data, takes about 10 seconds. For larger data, can take much longer
# because this demands lots of resources. 
tse_dmn &lt;- runDMN(tse, name = &quot;DMN&quot;, k = 1:7)

# It is stored in metadata
tse_dmn

## class: TreeSummarizedExperiment 
## dim: 151 27 
## metadata(1): DMN
## assays(3): counts relabundance clr
## rownames(151): 1726470 1726471 ... 17264756 17264757
## rowData names(6): Kingdom Phylum ... Family Genus
## colnames(27): A110 A12 ... A35 A38
## colData names(6): patient_status cohort ... Shannon_index Faith_diversity_index
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (151 rows)
## rowTree: 1 phylo tree(s) (151 leaves)
## colLinks: NULL
## colTree: NULL
</code></pre><p>Return information on metadata that the object contains.</p>
<pre><code>names(metadata(tse_dmn))

## [1] &quot;DMN&quot;
</code></pre><p>This returns a list of DMN objects for a closer investigation.</p>
<pre><code>getDMN(tse_dmn)

## [[1]]
## class: DMN 
## k: 1 
## samples x taxa: 27 x 151 
## Laplace: 12049.73 BIC: 12271.38 AIC: 12173.55 
## 
## [[2]]
## class: DMN 
## k: 2 
## samples x taxa: 27 x 151 
## Laplace: 11440.93 BIC: 12161.29 AIC: 11964.97 
## 
## [[3]]
## class: DMN 
## k: 3 
## samples x taxa: 27 x 151 
## Laplace: 11059.62 BIC: 12266.31 AIC: 11971.51 
## 
## [[4]]
## class: DMN 
## k: 4 
## samples x taxa: 27 x 151 
## Laplace: 11417.59 BIC: 13047.39 AIC: 12654.11 
## 
## [[5]]
## class: DMN 
## k: 5 
## samples x taxa: 27 x 151 
## Laplace: 11217.23 BIC: 13305.58 AIC: 12813.8 
## 
## [[6]]
## class: DMN 
## k: 6 
## samples x taxa: 27 x 151 
## Laplace: 11201.95 BIC: 13718.9 AIC: 13128.65 
## 
## [[7]]
## class: DMN 
## k: 7 
## samples x taxa: 27 x 151 
## Laplace: 11428.76 BIC: 14172.79 AIC: 13484.06
</code></pre><p>Show Laplace approximation (model evidence) for each model of the k
models.</p>
<pre><code>plotDMNFit(tse_dmn, type = &quot;laplace&quot;)
</code></pre><p><img src="dmm_files/figure-markdown_strict/unnamed-chunk-4-1.png" alt=""></p>
<p>Return the model that has the best fit.</p>
<pre><code>getBestDMNFit(tse_dmn, type = &quot;laplace&quot;)

## class: DMN 
## k: 3 
## samples x taxa: 27 x 151 
## Laplace: 11059.62 BIC: 12266.31 AIC: 11971.51
</code></pre><h2 id="pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors">PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</h2>
<p>Group samples and return DMNGroup object that contains a summary.
Patient status is used for grouping.</p>
<pre><code>dmn_group &lt;- calculateDMNgroup(tse_dmn, variable = &quot;patient_status&quot;, 
                             exprs_values = &quot;counts&quot;, k = 3)

dmn_group

## class: DMNGroup 
## summary:
##         k samples taxa      NLE    LogDet  Laplace      BIC      AIC
## ADHD    3      13  151 6018.131 -345.0838 5427.472 6601.657 6473.131
## Control 3      14  151 6647.269 -147.5224 6155.391 7247.655 7102.269
</code></pre><p>Mixture weights (rough measure of the cluster size).</p>
<pre><code>DirichletMultinomial::mixturewt(getBestDMNFit(tse_dmn))

##          pi    theta
## 1 0.4814815 31.27744
## 2 0.2962963 47.34342
## 3 0.2222222 92.27435
</code></pre><p>Samples-cluster assignment probabilities / how probable it is that
sample belongs to each cluster</p>
<pre><code>head(DirichletMultinomial::mixture(getBestDMNFit(tse_dmn)))

##               [,1]          [,2]          [,3]
## A110  1.000000e+00 1.255555e-144 7.568144e-205
## A12  9.827063e-117  6.138625e-93  1.000000e+00
## A15   1.000000e+00 9.542035e-119 3.373932e-234
## A19  5.371238e-112 1.823633e-107  1.000000e+00
## A21   2.144911e-93  4.740702e-96  1.000000e+00
## A23   1.000000e+00 8.845683e-111 1.936989e-161
</code></pre><p>Contribution of each taxa to each component</p>
<pre><code>head(DirichletMultinomial::fitted(getBestDMNFit(tse_dmn)))

##                 [,1]        [,2]       [,3]
## 1726470  6.352105278 2.898724985 20.1893609
## 1726471  5.287770022 0.002047648  0.1532225
## 17264731 0.001248792 9.144067155  2.0112127
## 17264726 0.140479905 1.363548633  7.5894318
## 1726472  2.104164413 3.523249039  2.6656653
## 17264724 0.072365662 0.002047648  9.8545967
</code></pre><p>Get the assignment probabilities</p>
<pre><code>prob &lt;- DirichletMultinomial::mixture(getBestDMNFit(tse_dmn))
# Add column names
colnames(prob) &lt;- c(&quot;comp1&quot;, &quot;comp2&quot;, &quot;comp3&quot;)

# For each row, finds column that has the highest value. Then extract the column 
# names of highest values.
vec &lt;- colnames(prob)[max.col(prob,ties.method = &quot;first&quot;)]

# Creates a data frame that contains principal coordinates and DMM information
euclidean_dmm_pcoa_df &lt;- cbind(euclidean_pcoa_df,
                               dmm_component = vec)

# Creates a plot
euclidean_dmm_plot &lt;- ggplot(data = euclidean_dmm_pcoa_df, 
                             aes(x=pcoa1, y=pcoa2,
                                 color = dmm_component)) +
  geom_point() +
  labs(x = &quot;Coordinate 1&quot;,
       y = &quot;Coordinate 2&quot;,
       title = &quot;PCoA with Aitchison distances&quot;) +  
  theme(title = element_text(size = 12)) # makes titles smaller

euclidean_dmm_plot
</code></pre><p><img src="dmm_files/figure-markdown_strict/unnamed-chunk-10-1.png" alt=""></p>

                                
                                </section>
                            
    </div>
    <div class="search-results">
        <div class="has-results">
            
            <h1 class="search-results-title"><span class='search-results-count'></span> results matching "<span class='search-query'></span>"</h1>
            <ul class="search-results-list"></ul>
            
        </div>
        <div class="no-results">
            
            <h1 class="search-results-title">No results matching "<span class='search-query'></span>"</h1>
            
        </div>
    </div>
</div>

                        </div>
                    </div>
                
            </div>

            
                
                <a href="beta.html#further-resources" class="navigation navigation-prev " aria-label="Previous page: Further resources">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
                <a href="dmm.html#pcoa-for-asv-level-data-with-bray-curtis;-with-dmm-clusters-shown-with-colors" class="navigation navigation-next " aria-label="Next page: PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors">
                    <i class="fa fa-angle-right"></i>
                </a>
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"DMM analysis","level":"1.8","depth":1,"next":{"title":"PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors","level":"1.8.1","depth":2,"anchor":"#pcoa-for-asv-level-data-with-bray-curtis;-with-dmm-clusters-shown-with-colors","path":"dmm.md","ref":"dmm.md#pcoa-for-asv-level-data-with-bray-curtis;-with-dmm-clusters-shown-with-colors","articles":[]},"previous":{"title":"Further resources","level":"1.7.4","depth":2,"anchor":"#further-resources","path":"beta.md","ref":"beta.md#further-resources","articles":[]},"dir":"ltr"},"config":{"gitbook":"*","theme":"default","variables":{},"plugins":[],"pluginsConfig":{"highlight":{},"search":{},"lunr":{"maxIndexSize":1000000,"ignoreSpecialCharacters":false},"sharing":{"facebook":true,"twitter":true,"google":false,"weibo":false,"instapaper":false,"vk":false,"all":["facebook","google","twitter","weibo","instapaper"]},"fontsettings":{"theme":"white","family":"sans","size":2},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":false}},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"pdf":{"pageNumbers":true,"fontSize":12,"fontFamily":"Arial","paperSize":"a4","chapterMark":"pagebreak","pageBreaksBefore":"/","margin":{"right":62,"left":62,"top":56,"bottom":56}},"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"}},"file":{"path":"dmm.md","mtime":"2021-06-29T12:28:15.839Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2021-06-29T12:30:25.060Z"},"basePath":".","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="gitbook/gitbook.js"></script>
    <script src="gitbook/theme.js"></script>
    
        
        <script src="gitbook/gitbook-plugin-search/search-engine.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-search/search.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-lunr/lunr.min.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-lunr/search-lunr.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-sharing/buttons.js"></script>
        
    
        
        <script src="gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

