<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Importing microbiome data | Introduction to microbiome data science</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Importing microbiome data | Introduction to microbiome data science" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/course_2021_radboud" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Importing microbiome data | Introduction to microbiome data science" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="Leo Lahti, Tuomas Borman, and Henrik Eckermann" />


<meta name="date" content="2021-07-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="getting-started.html"/>
<link rel="next" href="microbiome-data-exploration.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to miaverse</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-miaverse-framework"><i class="fa fa-check"></i><b>1.1</b> The miaverse framework</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#learning-goals"><i class="fa fa-check"></i><b>1.2</b> Learning goals</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="program.html"><a href="program.html"><i class="fa fa-check"></i><b>2</b> Program</a><ul>
<li class="chapter" data-level="2.1" data-path="program.html"><a href="program.html#monday-12-july-from-raw-sequences-to-ecological-data-analysis"><i class="fa fa-check"></i><b>2.1</b> Monday 12 July: from raw sequences to ecological data analysis</a></li>
<li class="chapter" data-level="2.2" data-path="program.html"><a href="program.html#tuesday-13-july---alpha-diversity"><i class="fa fa-check"></i><b>2.2</b> Tuesday 13 July - Alpha diversity</a></li>
<li class="chapter" data-level="2.3" data-path="program.html"><a href="program.html#wednesday-14-july---beta-diversity"><i class="fa fa-check"></i><b>2.3</b> Wednesday 14 July - Beta diversity</a></li>
<li class="chapter" data-level="2.4" data-path="program.html"><a href="program.html#thursday-15-july---differential-abundance"><i class="fa fa-check"></i><b>2.4</b> Thursday 15 July - Differential abundance</a></li>
<li class="chapter" data-level="2.5" data-path="program.html"><a href="program.html#friday-16-july-presentations-closing"><i class="fa fa-check"></i><b>2.5</b> Friday 16 July: Presentations &amp; closing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a><ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#checklist-before-the-course"><i class="fa fa-check"></i><b>3.1</b> Checklist (before the course)</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#support-and-resources"><i class="fa fa-check"></i><b>3.2</b> Support and resources</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#installing-and-loading-the-required-r-packages"><i class="fa fa-check"></i><b>3.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html"><i class="fa fa-check"></i><b>4</b> Importing microbiome data</a><ul>
<li class="chapter" data-level="4.1" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#data-access"><i class="fa fa-check"></i><b>4.1</b> Data access</a></li>
<li class="chapter" data-level="4.2" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#investigate-the-r-data-object"><i class="fa fa-check"></i><b>4.2</b> Investigate the R data object</a><ul>
<li class="chapter" data-level="4.2.1" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#rowdata-taxonomic-information"><i class="fa fa-check"></i><b>4.2.1</b> rowData (taxonomic information)</a></li>
<li class="chapter" data-level="4.2.2" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#coldata-sample-information"><i class="fa fa-check"></i><b>4.2.2</b> colData (sample information)</a></li>
<li class="chapter" data-level="4.2.3" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#phylogenetic-tree-information"><i class="fa fa-check"></i><b>4.2.3</b> Phylogenetic tree information</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#further-examples"><i class="fa fa-check"></i><b>4.3</b> Further examples</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html"><i class="fa fa-check"></i><b>5</b> Microbiome data exploration</a><ul>
<li class="chapter" data-level="5.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#investigate"><i class="fa fa-check"></i><b>5.1</b> Investigate</a></li>
<li class="chapter" data-level="5.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#manipulate"><i class="fa fa-check"></i><b>5.2</b> Manipulate</a><ul>
<li class="chapter" data-level="5.2.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#transformations"><i class="fa fa-check"></i><b>5.2.1</b> Transformations</a></li>
<li class="chapter" data-level="5.2.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#aggregation"><i class="fa fa-check"></i><b>5.2.2</b> Aggregation</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#visualize"><i class="fa fa-check"></i><b>5.3</b> Visualize</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>6</b> Alpha diversity</a><ul>
<li class="chapter" data-level="6.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#visualization"><i class="fa fa-check"></i><b>6.1</b> Visualization</a></li>
<li class="chapter" data-level="6.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#statistical-testing-and-comparisons"><i class="fa fa-check"></i><b>6.2</b> Statistical testing and comparisons</a></li>
<li class="chapter" data-level="6.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#further-resources"><i class="fa fa-check"></i><b>6.3</b> Further resources</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>7</b> Beta diversity</a><ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#examples-of-pcoa-with-different-settings"><i class="fa fa-check"></i><b>7.1</b> Examples of PCoA with different settings</a><ul>
<li class="chapter" data-level="7.1.1" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-bray-curtis"><i class="fa fa-check"></i><b>7.1.1</b> PCoA for ASV-level data with Bray-Curtis</a></li>
<li class="chapter" data-level="7.1.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-aitchison-distance"><i class="fa fa-check"></i><b>7.1.2</b> PCoA for ASV-level data with Aitchison distance</a></li>
<li class="chapter" data-level="7.1.3" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-aggregated-to-phylum-level"><i class="fa fa-check"></i><b>7.1.3</b> PCoA aggregated to Phylum level</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#highlighting-external-variables-on-pcoa-plot"><i class="fa fa-check"></i><b>7.2</b> Highlighting external variables on PCoA plot</a><ul>
<li class="chapter" data-level="7.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-with-discrete-sample-grouping-variable-shown-with-colors"><i class="fa fa-check"></i><b>7.2.1</b> PCoA with discrete sample grouping variable shown with colors</a></li>
<li class="chapter" data-level="7.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-plot-with-continuous-variable"><i class="fa fa-check"></i><b>7.2.2</b> PCoA plot with continuous variable</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#estimating-associations-with-an-external-variable"><i class="fa fa-check"></i><b>7.3</b> Estimating associations with an external variable</a></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#further-resources-1"><i class="fa fa-check"></i><b>7.4</b> Further resources</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="community-typing.html"><a href="community-typing.html"><i class="fa fa-check"></i><b>8</b> Community typing</a><ul>
<li class="chapter" data-level="8.1" data-path="community-typing.html"><a href="community-typing.html#dirichlet-multinomial-mixture-model"><i class="fa fa-check"></i><b>8.1</b> Dirichlet-Multinomial Mixture Model</a></li>
<li class="chapter" data-level="8.2" data-path="community-typing.html"><a href="community-typing.html#pcoa-for-asv-level-data-with-bray-curtis-with-dmm-clusters-shown-with-colors"><i class="fa fa-check"></i><b>8.2</b> PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html"><i class="fa fa-check"></i><b>9</b> Differential abundance analysis</a><ul>
<li class="chapter" data-level="9.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#wilcoxon-test"><i class="fa fa-check"></i><b>9.1</b> Wilcoxon test</a></li>
<li class="chapter" data-level="9.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#deseq2"><i class="fa fa-check"></i><b>9.2</b> DESeq2</a></li>
<li class="chapter" data-level="9.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#ancom-bc"><i class="fa fa-check"></i><b>9.3</b> ANCOM-BC</a></li>
<li class="chapter" data-level="9.4" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-wilcoxon-test-and-deseq2"><i class="fa fa-check"></i><b>9.4</b> Comparison of Wilcoxon test and DESeq2</a></li>
<li class="chapter" data-level="9.5" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-abundance"><i class="fa fa-check"></i><b>9.5</b> Comparison of abundance</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="study-material.html"><a href="study-material.html"><i class="fa fa-check"></i><b>10</b> Study material</a><ul>
<li class="chapter" data-level="10.1" data-path="study-material.html"><a href="study-material.html#lecture-slides"><i class="fa fa-check"></i><b>10.1</b> Lecture slides</a></li>
<li class="chapter" data-level="10.2" data-path="study-material.html"><a href="study-material.html#example-solutions"><i class="fa fa-check"></i><b>10.2</b> Example solutions</a></li>
<li class="chapter" data-level="10.3" data-path="study-material.html"><a href="study-material.html#r-programming-resources"><i class="fa fa-check"></i><b>10.3</b> R programming resources</a></li>
<li class="chapter" data-level="10.4" data-path="study-material.html"><a href="study-material.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>10.4</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="10.5" data-path="study-material.html"><a href="study-material.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>10.5</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="10.6" data-path="study-material.html"><a href="study-material.html#further-reading"><i class="fa fa-check"></i><b>10.6</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html"><i class="fa fa-check"></i><b>11</b> Miscellaneous material</a><ul>
<li class="chapter" data-level="11.1" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html#shapiro-wilk-test"><i class="fa fa-check"></i><b>11.1</b> Shapiro-Wilk test</a></li>
<li class="chapter" data-level="11.2" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html#deseq-details"><i class="fa fa-check"></i><b>11.2</b> Deseq details</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to microbiome data science</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importing-microbiome-data" class="section level1">
<h1><span class="header-section-number">4</span> Importing microbiome data</h1>
<p>This section demonstrates how to import microbiome profiling data in R.</p>
<p>The <em>biom</em> format is a standard file format for microbiome data. Here,
we import <em>biom</em> data files into a specific data container (structure)
in R. Specifically, we use a <em>TreeSummarizedExperiment</em> (TSE) data
container. This provides the basis for the <em>miaverse</em> data science
framework.</p>
<p>We use example data from the following publication: Tengeler AC <em>et
al.</em> (2020) <a href="https://doi.org/10.1186/s40168-020-00816-x"><strong>Gut microbiota from persons with
attention-deficit/hyperactivity disorder affects the brain in
mice</strong></a>. Microbiome
8:44. In this study, mice are colonized with microbiota from
participants with ADHD (attention deficit hyperactivity disorder) and
healthy participants. The aim of the study was to assess whether the
mice display ADHD behaviors after being inoculated with ADHD
microbiota, suggesting a role of the microbiome in ADHD pathology.</p>
<div id="data-access" class="section level2">
<h2><span class="header-section-number">4.1</span> Data access</h2>
<p><strong>Downloading the data</strong> You can download the data from
<a href="https://github.com/microbiome/course_2021_radboud/tree/main/data">data</a>
subfolder.</p>
<p>The data set consists of 3 files:</p>
<ul>
<li>biom file: abundance table and taxonomy information</li>
<li>csv file: sample metadata</li>
<li>tree file: phylogenetic tree</li>
</ul>
<p>Store the data in your desired local directory (for instance, <em>data/</em> under the
working directory), and define source file paths</p>
<p>Now we can load the (biom) data into a SummarizedExperiment (SE) object.</p>
</div>
<div id="investigate-the-r-data-object" class="section level2">
<h2><span class="header-section-number">4.2</span> Investigate the R data object</h2>
<p>We have now imported the data set in R. Let us investigate its contents.</p>
<pre><code>## class: SummarizedExperiment 
## dim: 151 27 
## metadata(0):
## assays(1): counts
## rownames(151): 1726470 1726471 ... 17264756 17264757
## rowData names(6): taxonomy1 taxonomy2 ... taxonomy5 taxonomy6
## colnames(27): A110 A111 ... A38 A39
## colData names(0):</code></pre>
<p>The <code>assays</code> slot includes a list of abundance tables. The imported abundance table is named as “counts”.
Let us inspect only the first cols and rows.</p>
<pre><code>##           A110  A111  A12
## 1726470  17722 11630    0
## 1726471  12052     0 2679
## 17264731     0   970    0</code></pre>
<div id="rowdata-taxonomic-information" class="section level3">
<h3><span class="header-section-number">4.2.1</span> rowData (taxonomic information)</h3>
<p>The <code>rowdata</code> includes taxonomic information from the biom file. The <code>head()</code> command
shows just the beginning of the data table for an overview.</p>
<p><code>knitr::kable()</code> is for printing the information more nicely.</p>
<table style="width:100%;">
<colgroup>
<col width="7%" />
<col width="10%" />
<col width="15%" />
<col width="15%" />
<col width="17%" />
<col width="18%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">taxonomy1</th>
<th align="left">taxonomy2</th>
<th align="left">taxonomy3</th>
<th align="left">taxonomy4</th>
<th align="left">taxonomy5</th>
<th align="left">taxonomy6</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1726470</td>
<td align="left">"k__Bacteria</td>
<td align="left">p__Bacteroidetes</td>
<td align="left">c__Bacteroidia</td>
<td align="left">o__Bacteroidales</td>
<td align="left">f__Bacteroidaceae</td>
<td align="left">g__Bacteroides"</td>
</tr>
<tr class="even">
<td align="left">1726471</td>
<td align="left">"k__Bacteria</td>
<td align="left">p__Bacteroidetes</td>
<td align="left">c__Bacteroidia</td>
<td align="left">o__Bacteroidales</td>
<td align="left">f__Bacteroidaceae</td>
<td align="left">g__Bacteroides"</td>
</tr>
<tr class="odd">
<td align="left">17264731</td>
<td align="left">"k__Bacteria</td>
<td align="left">p__Bacteroidetes</td>
<td align="left">c__Bacteroidia</td>
<td align="left">o__Bacteroidales</td>
<td align="left">f__Porphyromonadaceae</td>
<td align="left">g__Parabacteroides"</td>
</tr>
<tr class="even">
<td align="left">17264726</td>
<td align="left">"k__Bacteria</td>
<td align="left">p__Bacteroidetes</td>
<td align="left">c__Bacteroidia</td>
<td align="left">o__Bacteroidales</td>
<td align="left">f__Bacteroidaceae</td>
<td align="left">g__Bacteroides"</td>
</tr>
<tr class="odd">
<td align="left">1726472</td>
<td align="left">"k__Bacteria</td>
<td align="left">p__Verrucomicrobia</td>
<td align="left">c__Verrucomicrobiae</td>
<td align="left">o__Verrucomicrobiales</td>
<td align="left">f__Verrucomicrobiaceae</td>
<td align="left">g__Akkermansia"</td>
</tr>
<tr class="even">
<td align="left">17264724</td>
<td align="left">"k__Bacteria</td>
<td align="left">p__Bacteroidetes</td>
<td align="left">c__Bacteroidia</td>
<td align="left">o__Bacteroidales</td>
<td align="left">f__Bacteroidaceae</td>
<td align="left">g__Bacteroides"</td>
</tr>
</tbody>
</table>
<p>These taxonomic rank names (column names) are not real rank names. Let’s replace them with real rank names.</p>
<p>In addition to that, the taxa names include, e.g., ’"k__’ before the name, so let’s
make them cleaner by removing them.</p>
<table>
<colgroup>
<col width="8%" />
<col width="8%" />
<col width="15%" />
<col width="16%" />
<col width="17%" />
<col width="18%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Kingdom</th>
<th align="left">Phylum</th>
<th align="left">Class</th>
<th align="left">Order</th>
<th align="left">Family</th>
<th align="left">Genus</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1726470</td>
<td align="left">Bacteria</td>
<td align="left">Bacteroidetes</td>
<td align="left">Bacteroidia</td>
<td align="left">Bacteroidales</td>
<td align="left">Bacteroidaceae</td>
<td align="left">Bacteroides</td>
</tr>
<tr class="even">
<td align="left">1726471</td>
<td align="left">Bacteria</td>
<td align="left">Bacteroidetes</td>
<td align="left">Bacteroidia</td>
<td align="left">Bacteroidales</td>
<td align="left">Bacteroidaceae</td>
<td align="left">Bacteroides</td>
</tr>
<tr class="odd">
<td align="left">17264731</td>
<td align="left">Bacteria</td>
<td align="left">Bacteroidetes</td>
<td align="left">Bacteroidia</td>
<td align="left">Bacteroidales</td>
<td align="left">Porphyromonadaceae</td>
<td align="left">Parabacteroides</td>
</tr>
<tr class="even">
<td align="left">17264726</td>
<td align="left">Bacteria</td>
<td align="left">Bacteroidetes</td>
<td align="left">Bacteroidia</td>
<td align="left">Bacteroidales</td>
<td align="left">Bacteroidaceae</td>
<td align="left">Bacteroides</td>
</tr>
<tr class="odd">
<td align="left">1726472</td>
<td align="left">Bacteria</td>
<td align="left">Verrucomicrobia</td>
<td align="left">Verrucomicrobiae</td>
<td align="left">Verrucomicrobiales</td>
<td align="left">Verrucomicrobiaceae</td>
<td align="left">Akkermansia</td>
</tr>
<tr class="even">
<td align="left">17264724</td>
<td align="left">Bacteria</td>
<td align="left">Bacteroidetes</td>
<td align="left">Bacteroidia</td>
<td align="left">Bacteroidales</td>
<td align="left">Bacteroidaceae</td>
<td align="left">Bacteroides</td>
</tr>
</tbody>
</table>
</div>
<div id="coldata-sample-information" class="section level3">
<h3><span class="header-section-number">4.2.2</span> colData (sample information)</h3>
<p>We notice that the imported biom file did not contain the sample meta data
yet, so it includes an empty data frame.</p>
<pre><code>## DataFrame with 6 rows and 0 columns</code></pre>
<p>Let us add a sample meta data file.</p>
<p>Now <code>colData</code> includes the sample metadata. Use kable to print it more nicely.</p>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">patient_status</th>
<th align="left">cohort</th>
<th align="left">patient_status_vs_cohort</th>
<th align="left">sample_name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A110</td>
<td align="left">ADHD</td>
<td align="left">Cohort_1</td>
<td align="left">ADHD_Cohort_1</td>
<td align="left">A110</td>
</tr>
<tr class="even">
<td align="left">A12</td>
<td align="left">ADHD</td>
<td align="left">Cohort_1</td>
<td align="left">ADHD_Cohort_1</td>
<td align="left">A12</td>
</tr>
<tr class="odd">
<td align="left">A15</td>
<td align="left">ADHD</td>
<td align="left">Cohort_1</td>
<td align="left">ADHD_Cohort_1</td>
<td align="left">A15</td>
</tr>
<tr class="even">
<td align="left">A19</td>
<td align="left">ADHD</td>
<td align="left">Cohort_1</td>
<td align="left">ADHD_Cohort_1</td>
<td align="left">A19</td>
</tr>
<tr class="odd">
<td align="left">A21</td>
<td align="left">ADHD</td>
<td align="left">Cohort_2</td>
<td align="left">ADHD_Cohort_2</td>
<td align="left">A21</td>
</tr>
<tr class="even">
<td align="left">A23</td>
<td align="left">ADHD</td>
<td align="left">Cohort_2</td>
<td align="left">ADHD_Cohort_2</td>
<td align="left">A23</td>
</tr>
</tbody>
</table>
</div>
<div id="phylogenetic-tree-information" class="section level3">
<h3><span class="header-section-number">4.2.3</span> Phylogenetic tree information</h3>
<p>Now, let’s add a phylogenetic tree.</p>
<p>The current data object, se, is a SummarizedExperiment object. This
does not include a slot for adding a phylogenetic tree. In order to do
this, we can convert the SE object to an extended TreeSummarizedExperiment
object which also includes a <code>rowTree</code> slot.</p>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 151 27 
## metadata(0):
## assays(1): counts
## rownames(151): 1726470 1726471 ... 17264756 17264757
## rowData names(6): Kingdom Phylum ... Family Genus
## colnames(27): A110 A12 ... A35 A38
## colData names(4): patient_status cohort patient_status_vs_cohort sample_name
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: NULL
## rowTree: NULL
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Next, let us read the tree data file and add it to the R data object (tse).</p>
<pre><code>## class: TreeSummarizedExperiment 
## dim: 151 27 
## metadata(0):
## assays(1): counts
## rownames(151): 1726470 1726471 ... 17264756 17264757
## rowData names(6): Kingdom Phylum ... Family Genus
## colnames(27): A110 A12 ... A35 A38
## colData names(4): patient_status cohort patient_status_vs_cohort sample_name
## reducedDimNames(0):
## mainExpName: NULL
## altExpNames(0):
## rowLinks: a LinkDataFrame (151 rows)
## rowTree: 1 phylo tree(s) (151 leaves)
## colLinks: NULL
## colTree: NULL</code></pre>
<p>Now <code>rowTree</code> includes a phylogenetic tree:</p>
</div>
</div>
<div id="further-examples" class="section level2">
<h2><span class="header-section-number">4.3</span> Further examples</h2>
<p>See the online book for more examples on <a href="https://microbiome.github.io/OMA/data-introduction.html#loading-experimental-microbiome-data">data exploration and manipulation</a>.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="getting-started.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="microbiome-data-exploration.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
